{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 338,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>ENST</th>\n",
       "      <th>ENSG</th>\n",
       "      <th>protein</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Uniprot</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>ENST00000646641</td>\n",
       "      <td>ENSG00000267534</td>\n",
       "      <td>s1pr2_human</td>\n",
       "      <td>S1PR2</td>\n",
       "      <td>O95136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>121</td>\n",
       "      <td>ENST00000547270</td>\n",
       "      <td>ENSG00000257138</td>\n",
       "      <td>t2r38_human</td>\n",
       "      <td>TAS2R38</td>\n",
       "      <td>P59533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>219</td>\n",
       "      <td>ENST00000390675</td>\n",
       "      <td>ENSG00000256436</td>\n",
       "      <td>t2r31_human</td>\n",
       "      <td>TAS2R31</td>\n",
       "      <td>P59538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>344</td>\n",
       "      <td>ENST00000539585</td>\n",
       "      <td>ENSG00000256188</td>\n",
       "      <td>t2r30_human</td>\n",
       "      <td>TAS2R30</td>\n",
       "      <td>P59541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>483</td>\n",
       "      <td>ENST00000538986</td>\n",
       "      <td>ENSG00000255837</td>\n",
       "      <td>t2r20_human</td>\n",
       "      <td>TAS2R20</td>\n",
       "      <td>P59543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>64029</td>\n",
       "      <td>ENST00000510937</td>\n",
       "      <td>ENSG00000226306</td>\n",
       "      <td>npy6r_human</td>\n",
       "      <td>NPY6R</td>\n",
       "      <td>Q99463</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>64030</td>\n",
       "      <td>ENST00000641193</td>\n",
       "      <td>ENSG00000279301</td>\n",
       "      <td>o2t11_human</td>\n",
       "      <td>OR2T11</td>\n",
       "      <td>Q8NH01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>397</th>\n",
       "      <td>64031</td>\n",
       "      <td>ENST00000641732</td>\n",
       "      <td>ENSG00000172146</td>\n",
       "      <td>or1a1_human</td>\n",
       "      <td>OR1A1</td>\n",
       "      <td>Q9P1Q5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>398</th>\n",
       "      <td>64032</td>\n",
       "      <td>ENST00000328890</td>\n",
       "      <td>ENSG00000183024</td>\n",
       "      <td>or1g1_human</td>\n",
       "      <td>OR1G1</td>\n",
       "      <td>P47890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399</th>\n",
       "      <td>64034</td>\n",
       "      <td>ENST00000466706</td>\n",
       "      <td>ENSG00000179073</td>\n",
       "      <td>taar3_human</td>\n",
       "      <td>TAAR3P</td>\n",
       "      <td>Q9P1P4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>400 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     index             ENST             ENSG      protein   SYMBOL Uniprot\n",
       "0        0  ENST00000646641  ENSG00000267534  s1pr2_human    S1PR2  O95136\n",
       "1      121  ENST00000547270  ENSG00000257138  t2r38_human  TAS2R38  P59533\n",
       "2      219  ENST00000390675  ENSG00000256436  t2r31_human  TAS2R31  P59538\n",
       "3      344  ENST00000539585  ENSG00000256188  t2r30_human  TAS2R30  P59541\n",
       "4      483  ENST00000538986  ENSG00000255837  t2r20_human  TAS2R20  P59543\n",
       "..     ...              ...              ...          ...      ...     ...\n",
       "395  64029  ENST00000510937  ENSG00000226306  npy6r_human    NPY6R  Q99463\n",
       "396  64030  ENST00000641193  ENSG00000279301  o2t11_human   OR2T11  Q8NH01\n",
       "397  64031  ENST00000641732  ENSG00000172146  or1a1_human    OR1A1  Q9P1Q5\n",
       "398  64032  ENST00000328890  ENSG00000183024  or1g1_human    OR1G1  P47890\n",
       "399  64034  ENST00000466706  ENSG00000179073  taar3_human   TAAR3P  Q9P1P4\n",
       "\n",
       "[400 rows x 6 columns]"
      ]
     },
     "execution_count": 338,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in all gene names\n",
    "gene_list_raw =pd.read_csv(\"230323_EST_ENSG_GENE_new.csv\")\n",
    "gene_list_raw.rename(columns={'gene':'SYMBOL'},inplace=True)\n",
    "gene_list_raw\n",
    "# uni_list = gene_list_raw['Uniprot'].tolist()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 318,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>locus</th>\n",
       "      <th>alleles</th>\n",
       "      <th>markerID</th>\n",
       "      <th>gene</th>\n",
       "      <th>annotation</th>\n",
       "      <th>call_stats</th>\n",
       "      <th>n_cases</th>\n",
       "      <th>n_controls</th>\n",
       "      <th>heritability</th>\n",
       "      <th>...</th>\n",
       "      <th>coding_description</th>\n",
       "      <th>category</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>AF.Cases</th>\n",
       "      <th>AF.Controls</th>\n",
       "      <th>Pvalue</th>\n",
       "      <th>vcf</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3026054.0</td>\n",
       "      <td>chr1:203128854</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>chr1:203128854_A/G</td>\n",
       "      <td>ADORA1</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...</td>\n",
       "      <td>720</td>\n",
       "      <td>394121.0</td>\n",
       "      <td>0.041393</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>13</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>-1.0044</td>\n",
       "      <td>7.2457</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.88975</td>\n",
       "      <td>1 203128854 . A G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3026055.0</td>\n",
       "      <td>chr1:203128854</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>chr1:203128854_A/G</td>\n",
       "      <td>ADORA1</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...</td>\n",
       "      <td>542</td>\n",
       "      <td>394299.0</td>\n",
       "      <td>0.017957</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>13</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>-1.0036</td>\n",
       "      <td>7.5946</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.89487</td>\n",
       "      <td>1 203128854 . A G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3026056.0</td>\n",
       "      <td>chr1:203128854</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>chr1:203128854_A/G</td>\n",
       "      <td>ADORA1</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...</td>\n",
       "      <td>180</td>\n",
       "      <td>394661.0</td>\n",
       "      <td>0.135690</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>13</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>-1.0016</td>\n",
       "      <td>12.2150</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.93465</td>\n",
       "      <td>1 203128854 . A G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3026057.0</td>\n",
       "      <td>chr1:203128854</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>chr1:203128854_A/G</td>\n",
       "      <td>ADORA1</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...</td>\n",
       "      <td>2901</td>\n",
       "      <td>391940.0</td>\n",
       "      <td>0.044139</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>13</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>-1.0123</td>\n",
       "      <td>2.9949</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.73536</td>\n",
       "      <td>1 203128854 . A G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3026058.0</td>\n",
       "      <td>chr1:203128854</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>chr1:203128854_A/G</td>\n",
       "      <td>ADORA1</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...</td>\n",
       "      <td>774</td>\n",
       "      <td>394067.0</td>\n",
       "      <td>0.049357</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>13</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>-1.0040</td>\n",
       "      <td>6.2632</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.87264</td>\n",
       "      <td>1 203128854 . A G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>681245</th>\n",
       "      <td>NaN</td>\n",
       "      <td>chr7:159144768</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>chr7:159144768_G/A</td>\n",
       "      <td>VIPR2</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...</td>\n",
       "      <td>4901</td>\n",
       "      <td>389940.0</td>\n",
       "      <td>0.027519</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter XXI Factors influencing health status ...</td>\n",
       "      <td>14</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>-1.0160</td>\n",
       "      <td>2.5756</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.69325</td>\n",
       "      <td>7 159144768 . G A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>681246</th>\n",
       "      <td>NaN</td>\n",
       "      <td>chr7:159144768</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>chr7:159144768_G/A</td>\n",
       "      <td>VIPR2</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...</td>\n",
       "      <td>10199</td>\n",
       "      <td>384642.0</td>\n",
       "      <td>0.009953</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter XXI Factors influencing health status ...</td>\n",
       "      <td>14</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>1.8544</td>\n",
       "      <td>1.6976</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.27466</td>\n",
       "      <td>7 159144768 . G A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>681247</th>\n",
       "      <td>NaN</td>\n",
       "      <td>chr7:159144768</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>chr7:159144768_G/A</td>\n",
       "      <td>VIPR2</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...</td>\n",
       "      <td>5762</td>\n",
       "      <td>389079.0</td>\n",
       "      <td>0.010603</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter XXI Factors influencing health status ...</td>\n",
       "      <td>14</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>-1.0163</td>\n",
       "      <td>2.2083</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.64536</td>\n",
       "      <td>7 159144768 . G A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>681248</th>\n",
       "      <td>NaN</td>\n",
       "      <td>chr7:159144768</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>chr7:159144768_G/A</td>\n",
       "      <td>VIPR2</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...</td>\n",
       "      <td>10132</td>\n",
       "      <td>384709.0</td>\n",
       "      <td>0.015240</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter XXI Factors influencing health status ...</td>\n",
       "      <td>14</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>-1.0272</td>\n",
       "      <td>1.6512</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.53389</td>\n",
       "      <td>7 159144768 . G A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>681249</th>\n",
       "      <td>NaN</td>\n",
       "      <td>chr7:159144768</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>chr7:159144768_G/A</td>\n",
       "      <td>VIPR2</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...</td>\n",
       "      <td>23056</td>\n",
       "      <td>371785.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter XXI Factors influencing health status ...</td>\n",
       "      <td>14</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>-1.0650</td>\n",
       "      <td>1.1942</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>0.37246</td>\n",
       "      <td>7 159144768 . G A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>681250 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Unnamed: 0           locus    alleles            markerID    gene  \\\n",
       "0        3026054.0  chr1:203128854  [\"A\",\"G\"]  chr1:203128854_A/G  ADORA1   \n",
       "1        3026055.0  chr1:203128854  [\"A\",\"G\"]  chr1:203128854_A/G  ADORA1   \n",
       "2        3026056.0  chr1:203128854  [\"A\",\"G\"]  chr1:203128854_A/G  ADORA1   \n",
       "3        3026057.0  chr1:203128854  [\"A\",\"G\"]  chr1:203128854_A/G  ADORA1   \n",
       "4        3026058.0  chr1:203128854  [\"A\",\"G\"]  chr1:203128854_A/G  ADORA1   \n",
       "...            ...             ...        ...                 ...     ...   \n",
       "681245         NaN  chr7:159144768  [\"G\",\"A\"]  chr7:159144768_G/A   VIPR2   \n",
       "681246         NaN  chr7:159144768  [\"G\",\"A\"]  chr7:159144768_G/A   VIPR2   \n",
       "681247         NaN  chr7:159144768  [\"G\",\"A\"]  chr7:159144768_G/A   VIPR2   \n",
       "681248         NaN  chr7:159144768  [\"G\",\"A\"]  chr7:159144768_G/A   VIPR2   \n",
       "681249         NaN  chr7:159144768  [\"G\",\"A\"]  chr7:159144768_G/A   VIPR2   \n",
       "\n",
       "       annotation                                         call_stats  n_cases  \\\n",
       "0        missense  {\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...      720   \n",
       "1        missense  {\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...      542   \n",
       "2        missense  {\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...      180   \n",
       "3        missense  {\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...     2901   \n",
       "4        missense  {\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...      774   \n",
       "...           ...                                                ...      ...   \n",
       "681245   missense  {\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...     4901   \n",
       "681246   missense  {\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...    10199   \n",
       "681247   missense  {\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...     5762   \n",
       "681248   missense  {\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...    10132   \n",
       "681249   missense  {\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...    23056   \n",
       "\n",
       "        n_controls  heritability  ... coding_description  \\\n",
       "0         394121.0      0.041393  ...                NaN   \n",
       "1         394299.0      0.017957  ...                NaN   \n",
       "2         394661.0      0.135690  ...                NaN   \n",
       "3         391940.0      0.044139  ...                NaN   \n",
       "4         394067.0      0.049357  ...                NaN   \n",
       "...            ...           ...  ...                ...   \n",
       "681245    389940.0      0.027519  ...                NaN   \n",
       "681246    384642.0      0.009953  ...                NaN   \n",
       "681247    389079.0      0.010603  ...                NaN   \n",
       "681248    384709.0      0.015240  ...                NaN   \n",
       "681249    371785.0      0.000000  ...                NaN   \n",
       "\n",
       "                                                 category  AC        AF  \\\n",
       "0       Chapter II Neoplasms | C15-C26 Malignant neopl...  13  0.000016   \n",
       "1       Chapter II Neoplasms | C15-C26 Malignant neopl...  13  0.000016   \n",
       "2       Chapter II Neoplasms | C15-C26 Malignant neopl...  13  0.000016   \n",
       "3       Chapter II Neoplasms | C15-C26 Malignant neopl...  13  0.000016   \n",
       "4       Chapter II Neoplasms | C15-C26 Malignant neopl...  13  0.000016   \n",
       "...                                                   ...  ..       ...   \n",
       "681245  Chapter XXI Factors influencing health status ...  14  0.000018   \n",
       "681246  Chapter XXI Factors influencing health status ...  14  0.000018   \n",
       "681247  Chapter XXI Factors influencing health status ...  14  0.000018   \n",
       "681248  Chapter XXI Factors influencing health status ...  14  0.000018   \n",
       "681249  Chapter XXI Factors influencing health status ...  14  0.000018   \n",
       "\n",
       "          BETA       SE  AF.Cases  AF.Controls   Pvalue                vcf  \n",
       "0      -1.0044   7.2457  0.000000     0.000016  0.88975  1 203128854 . A G  \n",
       "1      -1.0036   7.5946  0.000000     0.000016  0.89487  1 203128854 . A G  \n",
       "2      -1.0016  12.2150  0.000000     0.000016  0.93465  1 203128854 . A G  \n",
       "3      -1.0123   2.9949  0.000000     0.000017  0.73536  1 203128854 . A G  \n",
       "4      -1.0040   6.2632  0.000000     0.000016  0.87264  1 203128854 . A G  \n",
       "...        ...      ...       ...          ...      ...                ...  \n",
       "681245 -1.0160   2.5756  0.000000     0.000018  0.69325  7 159144768 . G A  \n",
       "681246  1.8544   1.6976  0.000049     0.000017  0.27466  7 159144768 . G A  \n",
       "681247 -1.0163   2.2083  0.000000     0.000018  0.64536  7 159144768 . G A  \n",
       "681248 -1.0272   1.6512  0.000000     0.000018  0.53389  7 159144768 . G A  \n",
       "681249 -1.0650   1.1942  0.000000     0.000019  0.37246  7 159144768 . G A  \n",
       "\n",
       "[681250 rows x 33 columns]"
      ]
     },
     "execution_count": 318,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "icd10_df = pd.read_csv('icd10_missense_68.csv')\n",
    "icd10_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 319,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "icd10_df['Allele'] = icd10_df['markerID'].str.split('/').str[-1]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 320,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>locus</th>\n",
       "      <th>alleles</th>\n",
       "      <th>markerID</th>\n",
       "      <th>gene</th>\n",
       "      <th>annotation</th>\n",
       "      <th>call_stats</th>\n",
       "      <th>n_cases</th>\n",
       "      <th>n_controls</th>\n",
       "      <th>heritability</th>\n",
       "      <th>...</th>\n",
       "      <th>category</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>AF.Cases</th>\n",
       "      <th>AF.Controls</th>\n",
       "      <th>Pvalue</th>\n",
       "      <th>vcf</th>\n",
       "      <th>Allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3026054.0</td>\n",
       "      <td>chr1:203128854</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>chr1:203128854_A/G</td>\n",
       "      <td>ADORA1</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...</td>\n",
       "      <td>720</td>\n",
       "      <td>394121.0</td>\n",
       "      <td>0.041393</td>\n",
       "      <td>...</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>13</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>-1.0044</td>\n",
       "      <td>7.2457</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.88975</td>\n",
       "      <td>1 203128854 . A G</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3026055.0</td>\n",
       "      <td>chr1:203128854</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>chr1:203128854_A/G</td>\n",
       "      <td>ADORA1</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...</td>\n",
       "      <td>542</td>\n",
       "      <td>394299.0</td>\n",
       "      <td>0.017957</td>\n",
       "      <td>...</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>13</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>-1.0036</td>\n",
       "      <td>7.5946</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.89487</td>\n",
       "      <td>1 203128854 . A G</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3026056.0</td>\n",
       "      <td>chr1:203128854</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>chr1:203128854_A/G</td>\n",
       "      <td>ADORA1</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...</td>\n",
       "      <td>180</td>\n",
       "      <td>394661.0</td>\n",
       "      <td>0.135690</td>\n",
       "      <td>...</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>13</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>-1.0016</td>\n",
       "      <td>12.2150</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.93465</td>\n",
       "      <td>1 203128854 . A G</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3026057.0</td>\n",
       "      <td>chr1:203128854</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>chr1:203128854_A/G</td>\n",
       "      <td>ADORA1</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...</td>\n",
       "      <td>2901</td>\n",
       "      <td>391940.0</td>\n",
       "      <td>0.044139</td>\n",
       "      <td>...</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>13</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>-1.0123</td>\n",
       "      <td>2.9949</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.73536</td>\n",
       "      <td>1 203128854 . A G</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3026058.0</td>\n",
       "      <td>chr1:203128854</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>chr1:203128854_A/G</td>\n",
       "      <td>ADORA1</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...</td>\n",
       "      <td>774</td>\n",
       "      <td>394067.0</td>\n",
       "      <td>0.049357</td>\n",
       "      <td>...</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>13</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>-1.0040</td>\n",
       "      <td>6.2632</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.87264</td>\n",
       "      <td>1 203128854 . A G</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>681245</th>\n",
       "      <td>NaN</td>\n",
       "      <td>chr7:159144768</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>chr7:159144768_G/A</td>\n",
       "      <td>VIPR2</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...</td>\n",
       "      <td>4901</td>\n",
       "      <td>389940.0</td>\n",
       "      <td>0.027519</td>\n",
       "      <td>...</td>\n",
       "      <td>Chapter XXI Factors influencing health status ...</td>\n",
       "      <td>14</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>-1.0160</td>\n",
       "      <td>2.5756</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.69325</td>\n",
       "      <td>7 159144768 . G A</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>681246</th>\n",
       "      <td>NaN</td>\n",
       "      <td>chr7:159144768</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>chr7:159144768_G/A</td>\n",
       "      <td>VIPR2</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...</td>\n",
       "      <td>10199</td>\n",
       "      <td>384642.0</td>\n",
       "      <td>0.009953</td>\n",
       "      <td>...</td>\n",
       "      <td>Chapter XXI Factors influencing health status ...</td>\n",
       "      <td>14</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>1.8544</td>\n",
       "      <td>1.6976</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.27466</td>\n",
       "      <td>7 159144768 . G A</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>681247</th>\n",
       "      <td>NaN</td>\n",
       "      <td>chr7:159144768</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>chr7:159144768_G/A</td>\n",
       "      <td>VIPR2</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...</td>\n",
       "      <td>5762</td>\n",
       "      <td>389079.0</td>\n",
       "      <td>0.010603</td>\n",
       "      <td>...</td>\n",
       "      <td>Chapter XXI Factors influencing health status ...</td>\n",
       "      <td>14</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>-1.0163</td>\n",
       "      <td>2.2083</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.64536</td>\n",
       "      <td>7 159144768 . G A</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>681248</th>\n",
       "      <td>NaN</td>\n",
       "      <td>chr7:159144768</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>chr7:159144768_G/A</td>\n",
       "      <td>VIPR2</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...</td>\n",
       "      <td>10132</td>\n",
       "      <td>384709.0</td>\n",
       "      <td>0.015240</td>\n",
       "      <td>...</td>\n",
       "      <td>Chapter XXI Factors influencing health status ...</td>\n",
       "      <td>14</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>-1.0272</td>\n",
       "      <td>1.6512</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.53389</td>\n",
       "      <td>7 159144768 . G A</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>681249</th>\n",
       "      <td>NaN</td>\n",
       "      <td>chr7:159144768</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>chr7:159144768_G/A</td>\n",
       "      <td>VIPR2</td>\n",
       "      <td>missense</td>\n",
       "      <td>{\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...</td>\n",
       "      <td>23056</td>\n",
       "      <td>371785.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>Chapter XXI Factors influencing health status ...</td>\n",
       "      <td>14</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>-1.0650</td>\n",
       "      <td>1.1942</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>0.37246</td>\n",
       "      <td>7 159144768 . G A</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>681250 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Unnamed: 0           locus    alleles            markerID    gene  \\\n",
       "0        3026054.0  chr1:203128854  [\"A\",\"G\"]  chr1:203128854_A/G  ADORA1   \n",
       "1        3026055.0  chr1:203128854  [\"A\",\"G\"]  chr1:203128854_A/G  ADORA1   \n",
       "2        3026056.0  chr1:203128854  [\"A\",\"G\"]  chr1:203128854_A/G  ADORA1   \n",
       "3        3026057.0  chr1:203128854  [\"A\",\"G\"]  chr1:203128854_A/G  ADORA1   \n",
       "4        3026058.0  chr1:203128854  [\"A\",\"G\"]  chr1:203128854_A/G  ADORA1   \n",
       "...            ...             ...        ...                 ...     ...   \n",
       "681245         NaN  chr7:159144768  [\"G\",\"A\"]  chr7:159144768_G/A   VIPR2   \n",
       "681246         NaN  chr7:159144768  [\"G\",\"A\"]  chr7:159144768_G/A   VIPR2   \n",
       "681247         NaN  chr7:159144768  [\"G\",\"A\"]  chr7:159144768_G/A   VIPR2   \n",
       "681248         NaN  chr7:159144768  [\"G\",\"A\"]  chr7:159144768_G/A   VIPR2   \n",
       "681249         NaN  chr7:159144768  [\"G\",\"A\"]  chr7:159144768_G/A   VIPR2   \n",
       "\n",
       "       annotation                                         call_stats  n_cases  \\\n",
       "0        missense  {\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...      720   \n",
       "1        missense  {\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...      542   \n",
       "2        missense  {\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...      180   \n",
       "3        missense  {\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...     2901   \n",
       "4        missense  {\"AC\":13,\"AF\":1.6462322808421618E-5,\"AN\":78968...      774   \n",
       "...           ...                                                ...      ...   \n",
       "681245   missense  {\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...     4901   \n",
       "681246   missense  {\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...    10199   \n",
       "681247   missense  {\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...     5762   \n",
       "681248   missense  {\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...    10132   \n",
       "681249   missense  {\"AC\":14,\"AF\":1.772874513409263E-5,\"AN\":789678...    23056   \n",
       "\n",
       "        n_controls  heritability  ...  \\\n",
       "0         394121.0      0.041393  ...   \n",
       "1         394299.0      0.017957  ...   \n",
       "2         394661.0      0.135690  ...   \n",
       "3         391940.0      0.044139  ...   \n",
       "4         394067.0      0.049357  ...   \n",
       "...            ...           ...  ...   \n",
       "681245    389940.0      0.027519  ...   \n",
       "681246    384642.0      0.009953  ...   \n",
       "681247    389079.0      0.010603  ...   \n",
       "681248    384709.0      0.015240  ...   \n",
       "681249    371785.0      0.000000  ...   \n",
       "\n",
       "                                                 category  AC        AF  \\\n",
       "0       Chapter II Neoplasms | C15-C26 Malignant neopl...  13  0.000016   \n",
       "1       Chapter II Neoplasms | C15-C26 Malignant neopl...  13  0.000016   \n",
       "2       Chapter II Neoplasms | C15-C26 Malignant neopl...  13  0.000016   \n",
       "3       Chapter II Neoplasms | C15-C26 Malignant neopl...  13  0.000016   \n",
       "4       Chapter II Neoplasms | C15-C26 Malignant neopl...  13  0.000016   \n",
       "...                                                   ...  ..       ...   \n",
       "681245  Chapter XXI Factors influencing health status ...  14  0.000018   \n",
       "681246  Chapter XXI Factors influencing health status ...  14  0.000018   \n",
       "681247  Chapter XXI Factors influencing health status ...  14  0.000018   \n",
       "681248  Chapter XXI Factors influencing health status ...  14  0.000018   \n",
       "681249  Chapter XXI Factors influencing health status ...  14  0.000018   \n",
       "\n",
       "          BETA       SE  AF.Cases  AF.Controls   Pvalue                vcf  \\\n",
       "0      -1.0044   7.2457  0.000000     0.000016  0.88975  1 203128854 . A G   \n",
       "1      -1.0036   7.5946  0.000000     0.000016  0.89487  1 203128854 . A G   \n",
       "2      -1.0016  12.2150  0.000000     0.000016  0.93465  1 203128854 . A G   \n",
       "3      -1.0123   2.9949  0.000000     0.000017  0.73536  1 203128854 . A G   \n",
       "4      -1.0040   6.2632  0.000000     0.000016  0.87264  1 203128854 . A G   \n",
       "...        ...      ...       ...          ...      ...                ...   \n",
       "681245 -1.0160   2.5756  0.000000     0.000018  0.69325  7 159144768 . G A   \n",
       "681246  1.8544   1.6976  0.000049     0.000017  0.27466  7 159144768 . G A   \n",
       "681247 -1.0163   2.2083  0.000000     0.000018  0.64536  7 159144768 . G A   \n",
       "681248 -1.0272   1.6512  0.000000     0.000018  0.53389  7 159144768 . G A   \n",
       "681249 -1.0650   1.1942  0.000000     0.000019  0.37246  7 159144768 . G A   \n",
       "\n",
       "        Allele  \n",
       "0            G  \n",
       "1            G  \n",
       "2            G  \n",
       "3            G  \n",
       "4            G  \n",
       "...        ...  \n",
       "681245       A  \n",
       "681246       A  \n",
       "681247       A  \n",
       "681248       A  \n",
       "681249       A  \n",
       "\n",
       "[681250 rows x 34 columns]"
      ]
     },
     "execution_count": 320,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "icd10_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 339,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/joannahench/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3444: DtypeWarning: Columns (16,21,27) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>...</th>\n",
       "      <th>phastCons17way_primate</th>\n",
       "      <th>phastCons17way_primate_rankscore</th>\n",
       "      <th>phastCons30way_mammalian</th>\n",
       "      <th>phastCons30way_mammalian_rankscore</th>\n",
       "      <th>phyloP100way_vertebrate</th>\n",
       "      <th>phyloP100way_vertebrate_rankscore</th>\n",
       "      <th>phyloP17way_primate</th>\n",
       "      <th>phyloP17way_primate_rankscore</th>\n",
       "      <th>phyloP30way_mammalian</th>\n",
       "      <th>phyloP30way_mammalian_rankscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000346454.7</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.991000</td>\n",
       "      <td>0.66497</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000362072.8</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.991000</td>\n",
       "      <td>0.66497</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000538967.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.991000</td>\n",
       "      <td>0.66497</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000542968.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.991000</td>\n",
       "      <td>0.66497</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000544518.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.991000</td>\n",
       "      <td>0.66497</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64839</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907399-114907399</td>\n",
       "      <td>A</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00001293461</td>\n",
       "      <td>enhancer</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64840</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000276198.6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.145000</td>\n",
       "      <td>0.20128</td>\n",
       "      <td>0.195000</td>\n",
       "      <td>0.22031</td>\n",
       "      <td>6.437000</td>\n",
       "      <td>0.73344</td>\n",
       "      <td>0.609000</td>\n",
       "      <td>0.47794</td>\n",
       "      <td>1.049000</td>\n",
       "      <td>0.51903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64841</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000371950.3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64842</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000371951.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.145000</td>\n",
       "      <td>0.20128</td>\n",
       "      <td>0.195000</td>\n",
       "      <td>0.22031</td>\n",
       "      <td>6.437000</td>\n",
       "      <td>0.73344</td>\n",
       "      <td>0.609000</td>\n",
       "      <td>0.47794</td>\n",
       "      <td>1.049000</td>\n",
       "      <td>0.51903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64843</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00001293461</td>\n",
       "      <td>enhancer</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>64844 rows × 648 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Uploaded_variation                Location Allele  \\\n",
       "0                       .  11:113412715-113412715      C   \n",
       "1                       .  11:113412715-113412715      C   \n",
       "2                       .  11:113412715-113412715      C   \n",
       "3                       .  11:113412715-113412715      C   \n",
       "4                       .  11:113412715-113412715      C   \n",
       "...                   ...                     ...    ...   \n",
       "64839                   .   X:114907399-114907399      A   \n",
       "64840                   .   X:114907411-114907411      C   \n",
       "64841                   .   X:114907411-114907411      C   \n",
       "64842                   .   X:114907411-114907411      C   \n",
       "64843                   .   X:114907411-114907411      C   \n",
       "\n",
       "                     Consequence    IMPACT SYMBOL             Gene  \\\n",
       "0               missense_variant  MODERATE   DRD2  ENSG00000149295   \n",
       "1               missense_variant  MODERATE   DRD2  ENSG00000149295   \n",
       "2               missense_variant  MODERATE   DRD2  ENSG00000149295   \n",
       "3               missense_variant  MODERATE   DRD2  ENSG00000149295   \n",
       "4               missense_variant  MODERATE   DRD2  ENSG00000149295   \n",
       "...                          ...       ...    ...              ...   \n",
       "64839  regulatory_region_variant  MODIFIER      -                -   \n",
       "64840           missense_variant  MODERATE  HTR2C  ENSG00000147246   \n",
       "64841        3_prime_UTR_variant  MODIFIER  HTR2C  ENSG00000147246   \n",
       "64842           missense_variant  MODERATE  HTR2C  ENSG00000147246   \n",
       "64843  regulatory_region_variant  MODIFIER      -                -   \n",
       "\n",
       "            Feature_type            Feature         BIOTYPE  ...  \\\n",
       "0             Transcript  ENST00000346454.7  protein_coding  ...   \n",
       "1             Transcript  ENST00000362072.8  protein_coding  ...   \n",
       "2             Transcript  ENST00000538967.5  protein_coding  ...   \n",
       "3             Transcript  ENST00000542968.5  protein_coding  ...   \n",
       "4             Transcript  ENST00000544518.5  protein_coding  ...   \n",
       "...                  ...                ...             ...  ...   \n",
       "64839  RegulatoryFeature    ENSR00001293461        enhancer  ...   \n",
       "64840         Transcript  ENST00000276198.6  protein_coding  ...   \n",
       "64841         Transcript  ENST00000371950.3  protein_coding  ...   \n",
       "64842         Transcript  ENST00000371951.5  protein_coding  ...   \n",
       "64843  RegulatoryFeature    ENSR00001293461        enhancer  ...   \n",
       "\n",
       "      phastCons17way_primate phastCons17way_primate_rankscore  \\\n",
       "0                   0.991000                          0.66497   \n",
       "1                   0.991000                          0.66497   \n",
       "2                   0.991000                          0.66497   \n",
       "3                   0.991000                          0.66497   \n",
       "4                   0.991000                          0.66497   \n",
       "...                      ...                              ...   \n",
       "64839                      -                                -   \n",
       "64840               0.145000                          0.20128   \n",
       "64841                      -                                -   \n",
       "64842               0.145000                          0.20128   \n",
       "64843                      -                                -   \n",
       "\n",
       "      phastCons30way_mammalian phastCons30way_mammalian_rankscore  \\\n",
       "0                     1.000000                            0.86279   \n",
       "1                     1.000000                            0.86279   \n",
       "2                     1.000000                            0.86279   \n",
       "3                     1.000000                            0.86279   \n",
       "4                     1.000000                            0.86279   \n",
       "...                        ...                                ...   \n",
       "64839                        -                                  -   \n",
       "64840                 0.195000                            0.22031   \n",
       "64841                        -                                  -   \n",
       "64842                 0.195000                            0.22031   \n",
       "64843                        -                                  -   \n",
       "\n",
       "      phyloP100way_vertebrate phyloP100way_vertebrate_rankscore  \\\n",
       "0                    8.017000                           0.88732   \n",
       "1                    8.017000                           0.88732   \n",
       "2                    8.017000                           0.88732   \n",
       "3                    8.017000                           0.88732   \n",
       "4                    8.017000                           0.88732   \n",
       "...                       ...                               ...   \n",
       "64839                       -                                 -   \n",
       "64840                6.437000                           0.73344   \n",
       "64841                       -                                 -   \n",
       "64842                6.437000                           0.73344   \n",
       "64843                       -                                 -   \n",
       "\n",
       "      phyloP17way_primate phyloP17way_primate_rankscore phyloP30way_mammalian  \\\n",
       "0                0.665000                       0.62972              1.138000   \n",
       "1                0.665000                       0.62972              1.138000   \n",
       "2                0.665000                       0.62972              1.138000   \n",
       "3                0.665000                       0.62972              1.138000   \n",
       "4                0.665000                       0.62972              1.138000   \n",
       "...                   ...                           ...                   ...   \n",
       "64839                   -                             -                     -   \n",
       "64840            0.609000                       0.47794              1.049000   \n",
       "64841                   -                             -                     -   \n",
       "64842            0.609000                       0.47794              1.049000   \n",
       "64843                   -                             -                     -   \n",
       "\n",
       "      phyloP30way_mammalian_rankscore  \n",
       "0                             0.64695  \n",
       "1                             0.64695  \n",
       "2                             0.64695  \n",
       "3                             0.64695  \n",
       "4                             0.64695  \n",
       "...                               ...  \n",
       "64839                               -  \n",
       "64840                         0.51903  \n",
       "64841                               -  \n",
       "64842                         0.51903  \n",
       "64843                               -  \n",
       "\n",
       "[64844 rows x 648 columns]"
      ]
     },
     "execution_count": 339,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "icd10_vep_score = pd.read_csv(\"gene68_icd10_mis_0323.txt\",sep='\\t')\n",
    "icd10_vep_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>...</th>\n",
       "      <th>phastCons17way_primate_rankscore</th>\n",
       "      <th>phastCons30way_mammalian</th>\n",
       "      <th>phastCons30way_mammalian_rankscore</th>\n",
       "      <th>phyloP100way_vertebrate</th>\n",
       "      <th>phyloP100way_vertebrate_rankscore</th>\n",
       "      <th>phyloP17way_primate</th>\n",
       "      <th>phyloP17way_primate_rankscore</th>\n",
       "      <th>phyloP30way_mammalian</th>\n",
       "      <th>phyloP30way_mammalian_rankscore</th>\n",
       "      <th>locus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000346454.7</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.66497</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000362072.8</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.66497</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000538967.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.66497</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000542968.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.66497</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000544518.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.66497</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64839</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907399-114907399</td>\n",
       "      <td>A</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00001293461</td>\n",
       "      <td>enhancer</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:114907399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64840</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000276198.6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.20128</td>\n",
       "      <td>0.195000</td>\n",
       "      <td>0.22031</td>\n",
       "      <td>6.437000</td>\n",
       "      <td>0.73344</td>\n",
       "      <td>0.609000</td>\n",
       "      <td>0.47794</td>\n",
       "      <td>1.049000</td>\n",
       "      <td>0.51903</td>\n",
       "      <td>chrX:114907411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64841</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000371950.3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:114907411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64842</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000371951.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.20128</td>\n",
       "      <td>0.195000</td>\n",
       "      <td>0.22031</td>\n",
       "      <td>6.437000</td>\n",
       "      <td>0.73344</td>\n",
       "      <td>0.609000</td>\n",
       "      <td>0.47794</td>\n",
       "      <td>1.049000</td>\n",
       "      <td>0.51903</td>\n",
       "      <td>chrX:114907411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64843</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00001293461</td>\n",
       "      <td>enhancer</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:114907411</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>64844 rows × 649 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Uploaded_variation                Location Allele  \\\n",
       "0                       .  11:113412715-113412715      C   \n",
       "1                       .  11:113412715-113412715      C   \n",
       "2                       .  11:113412715-113412715      C   \n",
       "3                       .  11:113412715-113412715      C   \n",
       "4                       .  11:113412715-113412715      C   \n",
       "...                   ...                     ...    ...   \n",
       "64839                   .   X:114907399-114907399      A   \n",
       "64840                   .   X:114907411-114907411      C   \n",
       "64841                   .   X:114907411-114907411      C   \n",
       "64842                   .   X:114907411-114907411      C   \n",
       "64843                   .   X:114907411-114907411      C   \n",
       "\n",
       "                     Consequence    IMPACT SYMBOL             Gene  \\\n",
       "0               missense_variant  MODERATE   DRD2  ENSG00000149295   \n",
       "1               missense_variant  MODERATE   DRD2  ENSG00000149295   \n",
       "2               missense_variant  MODERATE   DRD2  ENSG00000149295   \n",
       "3               missense_variant  MODERATE   DRD2  ENSG00000149295   \n",
       "4               missense_variant  MODERATE   DRD2  ENSG00000149295   \n",
       "...                          ...       ...    ...              ...   \n",
       "64839  regulatory_region_variant  MODIFIER      -                -   \n",
       "64840           missense_variant  MODERATE  HTR2C  ENSG00000147246   \n",
       "64841        3_prime_UTR_variant  MODIFIER  HTR2C  ENSG00000147246   \n",
       "64842           missense_variant  MODERATE  HTR2C  ENSG00000147246   \n",
       "64843  regulatory_region_variant  MODIFIER      -                -   \n",
       "\n",
       "            Feature_type            Feature         BIOTYPE  ...  \\\n",
       "0             Transcript  ENST00000346454.7  protein_coding  ...   \n",
       "1             Transcript  ENST00000362072.8  protein_coding  ...   \n",
       "2             Transcript  ENST00000538967.5  protein_coding  ...   \n",
       "3             Transcript  ENST00000542968.5  protein_coding  ...   \n",
       "4             Transcript  ENST00000544518.5  protein_coding  ...   \n",
       "...                  ...                ...             ...  ...   \n",
       "64839  RegulatoryFeature    ENSR00001293461        enhancer  ...   \n",
       "64840         Transcript  ENST00000276198.6  protein_coding  ...   \n",
       "64841         Transcript  ENST00000371950.3  protein_coding  ...   \n",
       "64842         Transcript  ENST00000371951.5  protein_coding  ...   \n",
       "64843  RegulatoryFeature    ENSR00001293461        enhancer  ...   \n",
       "\n",
       "      phastCons17way_primate_rankscore phastCons30way_mammalian  \\\n",
       "0                              0.66497                 1.000000   \n",
       "1                              0.66497                 1.000000   \n",
       "2                              0.66497                 1.000000   \n",
       "3                              0.66497                 1.000000   \n",
       "4                              0.66497                 1.000000   \n",
       "...                                ...                      ...   \n",
       "64839                                -                        -   \n",
       "64840                          0.20128                 0.195000   \n",
       "64841                                -                        -   \n",
       "64842                          0.20128                 0.195000   \n",
       "64843                                -                        -   \n",
       "\n",
       "      phastCons30way_mammalian_rankscore phyloP100way_vertebrate  \\\n",
       "0                                0.86279                8.017000   \n",
       "1                                0.86279                8.017000   \n",
       "2                                0.86279                8.017000   \n",
       "3                                0.86279                8.017000   \n",
       "4                                0.86279                8.017000   \n",
       "...                                  ...                     ...   \n",
       "64839                                  -                       -   \n",
       "64840                            0.22031                6.437000   \n",
       "64841                                  -                       -   \n",
       "64842                            0.22031                6.437000   \n",
       "64843                                  -                       -   \n",
       "\n",
       "      phyloP100way_vertebrate_rankscore phyloP17way_primate  \\\n",
       "0                               0.88732            0.665000   \n",
       "1                               0.88732            0.665000   \n",
       "2                               0.88732            0.665000   \n",
       "3                               0.88732            0.665000   \n",
       "4                               0.88732            0.665000   \n",
       "...                                 ...                 ...   \n",
       "64839                                 -                   -   \n",
       "64840                           0.73344            0.609000   \n",
       "64841                                 -                   -   \n",
       "64842                           0.73344            0.609000   \n",
       "64843                                 -                   -   \n",
       "\n",
       "      phyloP17way_primate_rankscore phyloP30way_mammalian  \\\n",
       "0                           0.62972              1.138000   \n",
       "1                           0.62972              1.138000   \n",
       "2                           0.62972              1.138000   \n",
       "3                           0.62972              1.138000   \n",
       "4                           0.62972              1.138000   \n",
       "...                             ...                   ...   \n",
       "64839                             -                     -   \n",
       "64840                       0.47794              1.049000   \n",
       "64841                             -                     -   \n",
       "64842                       0.47794              1.049000   \n",
       "64843                             -                     -   \n",
       "\n",
       "      phyloP30way_mammalian_rankscore            locus  \n",
       "0                             0.64695  chr11:113412715  \n",
       "1                             0.64695  chr11:113412715  \n",
       "2                             0.64695  chr11:113412715  \n",
       "3                             0.64695  chr11:113412715  \n",
       "4                             0.64695  chr11:113412715  \n",
       "...                               ...              ...  \n",
       "64839                               -   chrX:114907399  \n",
       "64840                         0.51903   chrX:114907411  \n",
       "64841                               -   chrX:114907411  \n",
       "64842                         0.51903   chrX:114907411  \n",
       "64843                               -   chrX:114907411  \n",
       "\n",
       "[64844 rows x 649 columns]"
      ]
     },
     "execution_count": 340,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "location = icd10_vep_score['Location'].str.split('-',expand=True)[0]\n",
    "\n",
    "icd10_vep_score['locus']='chr'+location\n",
    "icd10_vep_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 341,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>...</th>\n",
       "      <th>phastCons30way_mammalian</th>\n",
       "      <th>phastCons30way_mammalian_rankscore</th>\n",
       "      <th>phyloP100way_vertebrate</th>\n",
       "      <th>phyloP100way_vertebrate_rankscore</th>\n",
       "      <th>phyloP17way_primate</th>\n",
       "      <th>phyloP17way_primate_rankscore</th>\n",
       "      <th>phyloP30way_mammalian</th>\n",
       "      <th>phyloP30way_mammalian_rankscore</th>\n",
       "      <th>locus</th>\n",
       "      <th>ENST_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000346454.7</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>ENST00000346454</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000362072.8</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>ENST00000362072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000538967.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>ENST00000538967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000542968.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>ENST00000542968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000544518.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>ENST00000544518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64839</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907399-114907399</td>\n",
       "      <td>A</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00001293461</td>\n",
       "      <td>enhancer</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:114907399</td>\n",
       "      <td>ENSR00001293461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64840</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000276198.6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.195000</td>\n",
       "      <td>0.22031</td>\n",
       "      <td>6.437000</td>\n",
       "      <td>0.73344</td>\n",
       "      <td>0.609000</td>\n",
       "      <td>0.47794</td>\n",
       "      <td>1.049000</td>\n",
       "      <td>0.51903</td>\n",
       "      <td>chrX:114907411</td>\n",
       "      <td>ENST00000276198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64841</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000371950.3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:114907411</td>\n",
       "      <td>ENST00000371950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64842</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000371951.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.195000</td>\n",
       "      <td>0.22031</td>\n",
       "      <td>6.437000</td>\n",
       "      <td>0.73344</td>\n",
       "      <td>0.609000</td>\n",
       "      <td>0.47794</td>\n",
       "      <td>1.049000</td>\n",
       "      <td>0.51903</td>\n",
       "      <td>chrX:114907411</td>\n",
       "      <td>ENST00000371951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64843</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00001293461</td>\n",
       "      <td>enhancer</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:114907411</td>\n",
       "      <td>ENSR00001293461</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>64844 rows × 650 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Uploaded_variation                Location Allele  \\\n",
       "0                       .  11:113412715-113412715      C   \n",
       "1                       .  11:113412715-113412715      C   \n",
       "2                       .  11:113412715-113412715      C   \n",
       "3                       .  11:113412715-113412715      C   \n",
       "4                       .  11:113412715-113412715      C   \n",
       "...                   ...                     ...    ...   \n",
       "64839                   .   X:114907399-114907399      A   \n",
       "64840                   .   X:114907411-114907411      C   \n",
       "64841                   .   X:114907411-114907411      C   \n",
       "64842                   .   X:114907411-114907411      C   \n",
       "64843                   .   X:114907411-114907411      C   \n",
       "\n",
       "                     Consequence    IMPACT SYMBOL             Gene  \\\n",
       "0               missense_variant  MODERATE   DRD2  ENSG00000149295   \n",
       "1               missense_variant  MODERATE   DRD2  ENSG00000149295   \n",
       "2               missense_variant  MODERATE   DRD2  ENSG00000149295   \n",
       "3               missense_variant  MODERATE   DRD2  ENSG00000149295   \n",
       "4               missense_variant  MODERATE   DRD2  ENSG00000149295   \n",
       "...                          ...       ...    ...              ...   \n",
       "64839  regulatory_region_variant  MODIFIER      -                -   \n",
       "64840           missense_variant  MODERATE  HTR2C  ENSG00000147246   \n",
       "64841        3_prime_UTR_variant  MODIFIER  HTR2C  ENSG00000147246   \n",
       "64842           missense_variant  MODERATE  HTR2C  ENSG00000147246   \n",
       "64843  regulatory_region_variant  MODIFIER      -                -   \n",
       "\n",
       "            Feature_type            Feature         BIOTYPE  ...  \\\n",
       "0             Transcript  ENST00000346454.7  protein_coding  ...   \n",
       "1             Transcript  ENST00000362072.8  protein_coding  ...   \n",
       "2             Transcript  ENST00000538967.5  protein_coding  ...   \n",
       "3             Transcript  ENST00000542968.5  protein_coding  ...   \n",
       "4             Transcript  ENST00000544518.5  protein_coding  ...   \n",
       "...                  ...                ...             ...  ...   \n",
       "64839  RegulatoryFeature    ENSR00001293461        enhancer  ...   \n",
       "64840         Transcript  ENST00000276198.6  protein_coding  ...   \n",
       "64841         Transcript  ENST00000371950.3  protein_coding  ...   \n",
       "64842         Transcript  ENST00000371951.5  protein_coding  ...   \n",
       "64843  RegulatoryFeature    ENSR00001293461        enhancer  ...   \n",
       "\n",
       "      phastCons30way_mammalian phastCons30way_mammalian_rankscore  \\\n",
       "0                     1.000000                            0.86279   \n",
       "1                     1.000000                            0.86279   \n",
       "2                     1.000000                            0.86279   \n",
       "3                     1.000000                            0.86279   \n",
       "4                     1.000000                            0.86279   \n",
       "...                        ...                                ...   \n",
       "64839                        -                                  -   \n",
       "64840                 0.195000                            0.22031   \n",
       "64841                        -                                  -   \n",
       "64842                 0.195000                            0.22031   \n",
       "64843                        -                                  -   \n",
       "\n",
       "      phyloP100way_vertebrate phyloP100way_vertebrate_rankscore  \\\n",
       "0                    8.017000                           0.88732   \n",
       "1                    8.017000                           0.88732   \n",
       "2                    8.017000                           0.88732   \n",
       "3                    8.017000                           0.88732   \n",
       "4                    8.017000                           0.88732   \n",
       "...                       ...                               ...   \n",
       "64839                       -                                 -   \n",
       "64840                6.437000                           0.73344   \n",
       "64841                       -                                 -   \n",
       "64842                6.437000                           0.73344   \n",
       "64843                       -                                 -   \n",
       "\n",
       "      phyloP17way_primate phyloP17way_primate_rankscore phyloP30way_mammalian  \\\n",
       "0                0.665000                       0.62972              1.138000   \n",
       "1                0.665000                       0.62972              1.138000   \n",
       "2                0.665000                       0.62972              1.138000   \n",
       "3                0.665000                       0.62972              1.138000   \n",
       "4                0.665000                       0.62972              1.138000   \n",
       "...                   ...                           ...                   ...   \n",
       "64839                   -                             -                     -   \n",
       "64840            0.609000                       0.47794              1.049000   \n",
       "64841                   -                             -                     -   \n",
       "64842            0.609000                       0.47794              1.049000   \n",
       "64843                   -                             -                     -   \n",
       "\n",
       "      phyloP30way_mammalian_rankscore            locus          ENST_id  \n",
       "0                             0.64695  chr11:113412715  ENST00000346454  \n",
       "1                             0.64695  chr11:113412715  ENST00000362072  \n",
       "2                             0.64695  chr11:113412715  ENST00000538967  \n",
       "3                             0.64695  chr11:113412715  ENST00000542968  \n",
       "4                             0.64695  chr11:113412715  ENST00000544518  \n",
       "...                               ...              ...              ...  \n",
       "64839                               -   chrX:114907399  ENSR00001293461  \n",
       "64840                         0.51903   chrX:114907411  ENST00000276198  \n",
       "64841                               -   chrX:114907411  ENST00000371950  \n",
       "64842                         0.51903   chrX:114907411  ENST00000371951  \n",
       "64843                               -   chrX:114907411  ENSR00001293461  \n",
       "\n",
       "[64844 rows x 650 columns]"
      ]
     },
     "execution_count": 341,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# select the rows based on Feature(ENST id)\n",
    "icd10_vep_score['ENST_id'] = icd10_vep_score['Feature'].str.split('.',expand=True)[0]\n",
    "icd10_vep_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 342,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>...</th>\n",
       "      <th>phastCons30way_mammalian</th>\n",
       "      <th>phastCons30way_mammalian_rankscore</th>\n",
       "      <th>phyloP100way_vertebrate</th>\n",
       "      <th>phyloP100way_vertebrate_rankscore</th>\n",
       "      <th>phyloP17way_primate</th>\n",
       "      <th>phyloP17way_primate_rankscore</th>\n",
       "      <th>phyloP30way_mammalian</th>\n",
       "      <th>phyloP30way_mammalian_rankscore</th>\n",
       "      <th>locus</th>\n",
       "      <th>ENST_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000346454.7</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>ENST00000346454</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000362072.8</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>ENST00000362072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000538967.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>ENST00000538967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000542968.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>ENST00000542968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000544518.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>ENST00000544518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64834</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907387-114907387</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000371951.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>3.163000</td>\n",
       "      <td>0.50462</td>\n",
       "      <td>0.609000</td>\n",
       "      <td>0.47794</td>\n",
       "      <td>1.049000</td>\n",
       "      <td>0.51903</td>\n",
       "      <td>chrX:114907387</td>\n",
       "      <td>ENST00000371951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64836</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907399-114907399</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000276198.6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.969000</td>\n",
       "      <td>0.44821</td>\n",
       "      <td>3.888000</td>\n",
       "      <td>0.55917</td>\n",
       "      <td>0.618000</td>\n",
       "      <td>0.50648</td>\n",
       "      <td>1.083000</td>\n",
       "      <td>0.53824</td>\n",
       "      <td>chrX:114907399</td>\n",
       "      <td>ENST00000276198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64838</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907399-114907399</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000371951.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.969000</td>\n",
       "      <td>0.44821</td>\n",
       "      <td>3.888000</td>\n",
       "      <td>0.55917</td>\n",
       "      <td>0.618000</td>\n",
       "      <td>0.50648</td>\n",
       "      <td>1.083000</td>\n",
       "      <td>0.53824</td>\n",
       "      <td>chrX:114907399</td>\n",
       "      <td>ENST00000371951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64840</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000276198.6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.195000</td>\n",
       "      <td>0.22031</td>\n",
       "      <td>6.437000</td>\n",
       "      <td>0.73344</td>\n",
       "      <td>0.609000</td>\n",
       "      <td>0.47794</td>\n",
       "      <td>1.049000</td>\n",
       "      <td>0.51903</td>\n",
       "      <td>chrX:114907411</td>\n",
       "      <td>ENST00000276198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64842</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000371951.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.195000</td>\n",
       "      <td>0.22031</td>\n",
       "      <td>6.437000</td>\n",
       "      <td>0.73344</td>\n",
       "      <td>0.609000</td>\n",
       "      <td>0.47794</td>\n",
       "      <td>1.049000</td>\n",
       "      <td>0.51903</td>\n",
       "      <td>chrX:114907411</td>\n",
       "      <td>ENST00000371951</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>36905 rows × 650 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Uploaded_variation                Location Allele       Consequence  \\\n",
       "0                       .  11:113412715-113412715      C  missense_variant   \n",
       "1                       .  11:113412715-113412715      C  missense_variant   \n",
       "2                       .  11:113412715-113412715      C  missense_variant   \n",
       "3                       .  11:113412715-113412715      C  missense_variant   \n",
       "4                       .  11:113412715-113412715      C  missense_variant   \n",
       "...                   ...                     ...    ...               ...   \n",
       "64834                   .   X:114907387-114907387      C  missense_variant   \n",
       "64836                   .   X:114907399-114907399      A  missense_variant   \n",
       "64838                   .   X:114907399-114907399      A  missense_variant   \n",
       "64840                   .   X:114907411-114907411      C  missense_variant   \n",
       "64842                   .   X:114907411-114907411      C  missense_variant   \n",
       "\n",
       "         IMPACT SYMBOL             Gene Feature_type            Feature  \\\n",
       "0      MODERATE   DRD2  ENSG00000149295   Transcript  ENST00000346454.7   \n",
       "1      MODERATE   DRD2  ENSG00000149295   Transcript  ENST00000362072.8   \n",
       "2      MODERATE   DRD2  ENSG00000149295   Transcript  ENST00000538967.5   \n",
       "3      MODERATE   DRD2  ENSG00000149295   Transcript  ENST00000542968.5   \n",
       "4      MODERATE   DRD2  ENSG00000149295   Transcript  ENST00000544518.5   \n",
       "...         ...    ...              ...          ...                ...   \n",
       "64834  MODERATE  HTR2C  ENSG00000147246   Transcript  ENST00000371951.5   \n",
       "64836  MODERATE  HTR2C  ENSG00000147246   Transcript  ENST00000276198.6   \n",
       "64838  MODERATE  HTR2C  ENSG00000147246   Transcript  ENST00000371951.5   \n",
       "64840  MODERATE  HTR2C  ENSG00000147246   Transcript  ENST00000276198.6   \n",
       "64842  MODERATE  HTR2C  ENSG00000147246   Transcript  ENST00000371951.5   \n",
       "\n",
       "              BIOTYPE  ... phastCons30way_mammalian  \\\n",
       "0      protein_coding  ...                 1.000000   \n",
       "1      protein_coding  ...                 1.000000   \n",
       "2      protein_coding  ...                 1.000000   \n",
       "3      protein_coding  ...                 1.000000   \n",
       "4      protein_coding  ...                 1.000000   \n",
       "...               ...  ...                      ...   \n",
       "64834  protein_coding  ...                 1.000000   \n",
       "64836  protein_coding  ...                 0.969000   \n",
       "64838  protein_coding  ...                 0.969000   \n",
       "64840  protein_coding  ...                 0.195000   \n",
       "64842  protein_coding  ...                 0.195000   \n",
       "\n",
       "      phastCons30way_mammalian_rankscore phyloP100way_vertebrate  \\\n",
       "0                                0.86279                8.017000   \n",
       "1                                0.86279                8.017000   \n",
       "2                                0.86279                8.017000   \n",
       "3                                0.86279                8.017000   \n",
       "4                                0.86279                8.017000   \n",
       "...                                  ...                     ...   \n",
       "64834                            0.86279                3.163000   \n",
       "64836                            0.44821                3.888000   \n",
       "64838                            0.44821                3.888000   \n",
       "64840                            0.22031                6.437000   \n",
       "64842                            0.22031                6.437000   \n",
       "\n",
       "      phyloP100way_vertebrate_rankscore phyloP17way_primate  \\\n",
       "0                               0.88732            0.665000   \n",
       "1                               0.88732            0.665000   \n",
       "2                               0.88732            0.665000   \n",
       "3                               0.88732            0.665000   \n",
       "4                               0.88732            0.665000   \n",
       "...                                 ...                 ...   \n",
       "64834                           0.50462            0.609000   \n",
       "64836                           0.55917            0.618000   \n",
       "64838                           0.55917            0.618000   \n",
       "64840                           0.73344            0.609000   \n",
       "64842                           0.73344            0.609000   \n",
       "\n",
       "      phyloP17way_primate_rankscore phyloP30way_mammalian  \\\n",
       "0                           0.62972              1.138000   \n",
       "1                           0.62972              1.138000   \n",
       "2                           0.62972              1.138000   \n",
       "3                           0.62972              1.138000   \n",
       "4                           0.62972              1.138000   \n",
       "...                             ...                   ...   \n",
       "64834                       0.47794              1.049000   \n",
       "64836                       0.50648              1.083000   \n",
       "64838                       0.50648              1.083000   \n",
       "64840                       0.47794              1.049000   \n",
       "64842                       0.47794              1.049000   \n",
       "\n",
       "      phyloP30way_mammalian_rankscore            locus          ENST_id  \n",
       "0                             0.64695  chr11:113412715  ENST00000346454  \n",
       "1                             0.64695  chr11:113412715  ENST00000362072  \n",
       "2                             0.64695  chr11:113412715  ENST00000538967  \n",
       "3                             0.64695  chr11:113412715  ENST00000542968  \n",
       "4                             0.64695  chr11:113412715  ENST00000544518  \n",
       "...                               ...              ...              ...  \n",
       "64834                         0.51903   chrX:114907387  ENST00000371951  \n",
       "64836                         0.53824   chrX:114907399  ENST00000276198  \n",
       "64838                         0.53824   chrX:114907399  ENST00000371951  \n",
       "64840                         0.51903   chrX:114907411  ENST00000276198  \n",
       "64842                         0.51903   chrX:114907411  ENST00000371951  \n",
       "\n",
       "[36905 rows x 650 columns]"
      ]
     },
     "execution_count": 342,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# df1 = icd10_vep_score.drop_duplicates(subset=['Location','Allele'],keep='first')\n",
    "df1 = icd10_vep_score[icd10_vep_score['Consequence'].str.contains('missense')]\n",
    "# df1  = df1.drop(df1[df1['MutationTaster_AAE']=='-'].index)\n",
    "# df1 = icd10_vep_score.drop_duplicates(subset=['SYMBOL','Protein_position','Amino_acids'],keep='first')\n",
    "# df1 = icd10_vep_score.drop_duplicates(subset=['Location','Allele'],keep='first')\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 343,
   "metadata": {},
   "outputs": [],
   "source": [
    "df1 = df1[df1['ENST_id'].isin(gene_list_raw['ENST'].tolist())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 344,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>...</th>\n",
       "      <th>phastCons30way_mammalian</th>\n",
       "      <th>phastCons30way_mammalian_rankscore</th>\n",
       "      <th>phyloP100way_vertebrate</th>\n",
       "      <th>phyloP100way_vertebrate_rankscore</th>\n",
       "      <th>phyloP17way_primate</th>\n",
       "      <th>phyloP17way_primate_rankscore</th>\n",
       "      <th>phyloP30way_mammalian</th>\n",
       "      <th>phyloP30way_mammalian_rankscore</th>\n",
       "      <th>locus</th>\n",
       "      <th>ENST_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000362072.8</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>8.017000</td>\n",
       "      <td>0.88732</td>\n",
       "      <td>0.665000</td>\n",
       "      <td>0.62972</td>\n",
       "      <td>1.138000</td>\n",
       "      <td>0.64695</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>ENST00000362072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412762-113412762</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000362072.8</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.916000</td>\n",
       "      <td>0.38718</td>\n",
       "      <td>5.718000</td>\n",
       "      <td>0.68103</td>\n",
       "      <td>0.676000</td>\n",
       "      <td>0.76740</td>\n",
       "      <td>1.176000</td>\n",
       "      <td>0.78918</td>\n",
       "      <td>chr11:113412762</td>\n",
       "      <td>ENST00000362072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113412766-113412766</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000362072.8</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.772000</td>\n",
       "      <td>0.32703</td>\n",
       "      <td>0.409000</td>\n",
       "      <td>0.20803</td>\n",
       "      <td>0.676000</td>\n",
       "      <td>0.76740</td>\n",
       "      <td>0.189000</td>\n",
       "      <td>0.21855</td>\n",
       "      <td>chr11:113412766</td>\n",
       "      <td>ENST00000362072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>.</td>\n",
       "      <td>11:113416935-113416935</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>ENSG00000149295</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000362072.8</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.992000</td>\n",
       "      <td>0.54103</td>\n",
       "      <td>6.167000</td>\n",
       "      <td>0.71757</td>\n",
       "      <td>0.599000</td>\n",
       "      <td>0.40250</td>\n",
       "      <td>1.026000</td>\n",
       "      <td>0.45946</td>\n",
       "      <td>chr11:113416935</td>\n",
       "      <td>ENST00000362072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>.</td>\n",
       "      <td>13:77901095-77901095</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>EDNRB</td>\n",
       "      <td>ENSG00000136160</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000646607.2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.994000</td>\n",
       "      <td>0.56301</td>\n",
       "      <td>4.540000</td>\n",
       "      <td>0.60375</td>\n",
       "      <td>0.599000</td>\n",
       "      <td>0.40250</td>\n",
       "      <td>1.026000</td>\n",
       "      <td>0.45946</td>\n",
       "      <td>chr13:77901095</td>\n",
       "      <td>ENST00000646607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64824</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907377-114907377</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000276198.6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.998000</td>\n",
       "      <td>0.65909</td>\n",
       "      <td>1.396000</td>\n",
       "      <td>0.34140</td>\n",
       "      <td>0.549000</td>\n",
       "      <td>0.26987</td>\n",
       "      <td>0.947000</td>\n",
       "      <td>0.34431</td>\n",
       "      <td>chrX:114907377</td>\n",
       "      <td>ENST00000276198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64828</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907381-114907381</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000276198.6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>0.62031</td>\n",
       "      <td>2.041000</td>\n",
       "      <td>0.40834</td>\n",
       "      <td>0.549000</td>\n",
       "      <td>0.26987</td>\n",
       "      <td>0.947000</td>\n",
       "      <td>0.34431</td>\n",
       "      <td>chrX:114907381</td>\n",
       "      <td>ENST00000276198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64832</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907387-114907387</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000276198.6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.86279</td>\n",
       "      <td>3.163000</td>\n",
       "      <td>0.50462</td>\n",
       "      <td>0.609000</td>\n",
       "      <td>0.47794</td>\n",
       "      <td>1.049000</td>\n",
       "      <td>0.51903</td>\n",
       "      <td>chrX:114907387</td>\n",
       "      <td>ENST00000276198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64836</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907399-114907399</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000276198.6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.969000</td>\n",
       "      <td>0.44821</td>\n",
       "      <td>3.888000</td>\n",
       "      <td>0.55917</td>\n",
       "      <td>0.618000</td>\n",
       "      <td>0.50648</td>\n",
       "      <td>1.083000</td>\n",
       "      <td>0.53824</td>\n",
       "      <td>chrX:114907399</td>\n",
       "      <td>ENST00000276198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64840</th>\n",
       "      <td>.</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>ENSG00000147246</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000276198.6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.195000</td>\n",
       "      <td>0.22031</td>\n",
       "      <td>6.437000</td>\n",
       "      <td>0.73344</td>\n",
       "      <td>0.609000</td>\n",
       "      <td>0.47794</td>\n",
       "      <td>1.049000</td>\n",
       "      <td>0.51903</td>\n",
       "      <td>chrX:114907411</td>\n",
       "      <td>ENST00000276198</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15580 rows × 650 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Uploaded_variation                Location Allele       Consequence  \\\n",
       "1                       .  11:113412715-113412715      C  missense_variant   \n",
       "7                       .  11:113412762-113412762      C  missense_variant   \n",
       "13                      .  11:113412766-113412766      A  missense_variant   \n",
       "19                      .  11:113416935-113416935      T  missense_variant   \n",
       "29                      .    13:77901095-77901095      T  missense_variant   \n",
       "...                   ...                     ...    ...               ...   \n",
       "64824                   .   X:114907377-114907377      T  missense_variant   \n",
       "64828                   .   X:114907381-114907381      T  missense_variant   \n",
       "64832                   .   X:114907387-114907387      C  missense_variant   \n",
       "64836                   .   X:114907399-114907399      A  missense_variant   \n",
       "64840                   .   X:114907411-114907411      C  missense_variant   \n",
       "\n",
       "         IMPACT SYMBOL             Gene Feature_type            Feature  \\\n",
       "1      MODERATE   DRD2  ENSG00000149295   Transcript  ENST00000362072.8   \n",
       "7      MODERATE   DRD2  ENSG00000149295   Transcript  ENST00000362072.8   \n",
       "13     MODERATE   DRD2  ENSG00000149295   Transcript  ENST00000362072.8   \n",
       "19     MODERATE   DRD2  ENSG00000149295   Transcript  ENST00000362072.8   \n",
       "29     MODERATE  EDNRB  ENSG00000136160   Transcript  ENST00000646607.2   \n",
       "...         ...    ...              ...          ...                ...   \n",
       "64824  MODERATE  HTR2C  ENSG00000147246   Transcript  ENST00000276198.6   \n",
       "64828  MODERATE  HTR2C  ENSG00000147246   Transcript  ENST00000276198.6   \n",
       "64832  MODERATE  HTR2C  ENSG00000147246   Transcript  ENST00000276198.6   \n",
       "64836  MODERATE  HTR2C  ENSG00000147246   Transcript  ENST00000276198.6   \n",
       "64840  MODERATE  HTR2C  ENSG00000147246   Transcript  ENST00000276198.6   \n",
       "\n",
       "              BIOTYPE  ... phastCons30way_mammalian  \\\n",
       "1      protein_coding  ...                 1.000000   \n",
       "7      protein_coding  ...                 0.916000   \n",
       "13     protein_coding  ...                 0.772000   \n",
       "19     protein_coding  ...                 0.992000   \n",
       "29     protein_coding  ...                 0.994000   \n",
       "...               ...  ...                      ...   \n",
       "64824  protein_coding  ...                 0.998000   \n",
       "64828  protein_coding  ...                 0.997000   \n",
       "64832  protein_coding  ...                 1.000000   \n",
       "64836  protein_coding  ...                 0.969000   \n",
       "64840  protein_coding  ...                 0.195000   \n",
       "\n",
       "      phastCons30way_mammalian_rankscore phyloP100way_vertebrate  \\\n",
       "1                                0.86279                8.017000   \n",
       "7                                0.38718                5.718000   \n",
       "13                               0.32703                0.409000   \n",
       "19                               0.54103                6.167000   \n",
       "29                               0.56301                4.540000   \n",
       "...                                  ...                     ...   \n",
       "64824                            0.65909                1.396000   \n",
       "64828                            0.62031                2.041000   \n",
       "64832                            0.86279                3.163000   \n",
       "64836                            0.44821                3.888000   \n",
       "64840                            0.22031                6.437000   \n",
       "\n",
       "      phyloP100way_vertebrate_rankscore phyloP17way_primate  \\\n",
       "1                               0.88732            0.665000   \n",
       "7                               0.68103            0.676000   \n",
       "13                              0.20803            0.676000   \n",
       "19                              0.71757            0.599000   \n",
       "29                              0.60375            0.599000   \n",
       "...                                 ...                 ...   \n",
       "64824                           0.34140            0.549000   \n",
       "64828                           0.40834            0.549000   \n",
       "64832                           0.50462            0.609000   \n",
       "64836                           0.55917            0.618000   \n",
       "64840                           0.73344            0.609000   \n",
       "\n",
       "      phyloP17way_primate_rankscore phyloP30way_mammalian  \\\n",
       "1                           0.62972              1.138000   \n",
       "7                           0.76740              1.176000   \n",
       "13                          0.76740              0.189000   \n",
       "19                          0.40250              1.026000   \n",
       "29                          0.40250              1.026000   \n",
       "...                             ...                   ...   \n",
       "64824                       0.26987              0.947000   \n",
       "64828                       0.26987              0.947000   \n",
       "64832                       0.47794              1.049000   \n",
       "64836                       0.50648              1.083000   \n",
       "64840                       0.47794              1.049000   \n",
       "\n",
       "      phyloP30way_mammalian_rankscore            locus          ENST_id  \n",
       "1                             0.64695  chr11:113412715  ENST00000362072  \n",
       "7                             0.78918  chr11:113412762  ENST00000362072  \n",
       "13                            0.21855  chr11:113412766  ENST00000362072  \n",
       "19                            0.45946  chr11:113416935  ENST00000362072  \n",
       "29                            0.45946   chr13:77901095  ENST00000646607  \n",
       "...                               ...              ...              ...  \n",
       "64824                         0.34431   chrX:114907377  ENST00000276198  \n",
       "64828                         0.34431   chrX:114907381  ENST00000276198  \n",
       "64832                         0.51903   chrX:114907387  ENST00000276198  \n",
       "64836                         0.53824   chrX:114907399  ENST00000276198  \n",
       "64840                         0.51903   chrX:114907411  ENST00000276198  \n",
       "\n",
       "[15580 rows x 650 columns]"
      ]
     },
     "execution_count": 344,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1 = df1.drop_duplicates(subset=['Location','Allele'],keep='first')\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['missense_variant', 'missense_variant,splice_region_variant'],\n",
       "      dtype='<U38')"
      ]
     },
     "execution_count": 345,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.unique(df1['Consequence'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 346,
   "metadata": {},
   "outputs": [],
   "source": [
    "# missense mutation output \n",
    "df1.to_csv('icd10_missense_0323.csv',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "68"
      ]
     },
     "execution_count": 309,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(np.unique(df1['SYMBOL']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### rankscore list for correlation calculation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # rankscore list\n",
    "# # import the names of VEPs\n",
    "# VEP_list = [\"SIFT\",\n",
    "# \"PolyPhen\",\n",
    "# \"BayesDel_addAF_rankscore\",\n",
    "# \"BayesDel_noAF_rankscore\",\n",
    "# \"CADD_raw_rankscore\",\n",
    "# \"CADD_raw_rankscore_hg19\",\n",
    "# \"ClinPred_rankscore\",\n",
    "# \"DANN_rankscore\",\n",
    "# \"DEOGEN2_rankscore\",\n",
    "# \"Eigen-PC-raw_coding_rankscore\",\n",
    "# \"Eigen-raw_coding_rankscore\",\n",
    "# \"FATHMM_converted_rankscore\",\n",
    "# \"GERP++_RS_rankscore\",\n",
    "# \"GM12878_fitCons_rankscore\",\n",
    "# \"GenoCanyon_rankscore\",\n",
    "# \"H1-hESC_fitCons_rankscore\",\n",
    "# \"HUVEC_fitCons_rankscore\",\n",
    "# \"LIST-S2_rankscore\",\n",
    "# \"LRT_converted_rankscore\",\n",
    "# \"M-CAP_rankscore\",\n",
    "# \"MPC_rankscore\",\n",
    "# \"MVP_rankscore\",\n",
    "# \"MetaLR_rankscore\",\n",
    "# \"MetaRNN_rankscore\",\n",
    "# \"MetaSVM_rankscore\",\n",
    "# \"MutPred_rankscore\",\n",
    "# \"MutationAssessor_rankscore\",\n",
    "# \"MutationTaster_converted_rankscore\",\n",
    "# \"PROVEAN_converted_rankscore\",\n",
    "# \"PrimateAI_rankscore\",\n",
    "# \"REVEL_rankscore\",\n",
    "# \"SiPhy_29way_logOdds_rankscore\",\n",
    "# \"VEST4_rankscore\",\n",
    "# \"bStatistic_converted_rankscore\",\n",
    "# \"fathmm-MKL_coding_rankscore\",\n",
    "# \"fathmm-XF_coding_rankscore\",\n",
    "# \"integrated_fitCons_rankscore\",\n",
    "# \"phastCons100way_vertebrate_rankscore\",\n",
    "# \"phastCons17way_primate_rankscore\",\n",
    "# \"phastCons30way_mammalian_rankscore\",\n",
    "# \"phyloP100way_vertebrate_rankscore\",\n",
    "# \"phyloP17way_primate_rankscore\",\n",
    "# \"phyloP30way_mammalian_rankscore\"]"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### raw score for classification "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 347,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import the names of VEPs\n",
    "VEP_list = [\"SIFT\",\n",
    "\"PolyPhen\",\n",
    "\"BayesDel_addAF_score\",\n",
    "\"BayesDel_noAF_score\",\n",
    "\"CADD_phred\",\n",
    "\"ClinPred_score\",\n",
    "\"DANN_score\",\n",
    "\"DEOGEN2_score\",\n",
    "\"FATHMM_score\",\n",
    "\"integrated_fitCons_score\",\n",
    "\"LIST-S2_score\",\n",
    "\"LRT_score\",\n",
    "\"M-CAP_score\",\n",
    "\"MVP_score\",\n",
    "\"MetaLR_score\",\n",
    "\"MetaRNN_score\",\n",
    "\"MetaSVM_score\",\n",
    "\"MutationAssessor_score\",\n",
    "\"PROVEAN_score\",\n",
    "\"PrimateAI_score\",\n",
    "\"REVEL_score\",\n",
    "\"VEST4_score\",\n",
    "\"GERP++_RS\",\n",
    "\"SiPhy_29way_logOdds\",\n",
    "\"MutPred_score\"]"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### filter and merge with raw vep score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 348,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>locus</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BayesDel_addAF_score</th>\n",
       "      <th>BayesDel_noAF_score</th>\n",
       "      <th>...</th>\n",
       "      <th>MetaRNN_score</th>\n",
       "      <th>MetaSVM_score</th>\n",
       "      <th>MutationAssessor_score</th>\n",
       "      <th>PROVEAN_score</th>\n",
       "      <th>PrimateAI_score</th>\n",
       "      <th>REVEL_score</th>\n",
       "      <th>VEST4_score</th>\n",
       "      <th>GERP++_RS</th>\n",
       "      <th>SiPhy_29way_logOdds</th>\n",
       "      <th>MutPred_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>327</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>K/E</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>tolerated(0.27)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td>-0.260005</td>\n",
       "      <td>-0.140988</td>\n",
       "      <td>...</td>\n",
       "      <td>0.006436944,0.006436944,0.006436944,0.00643694...</td>\n",
       "      <td>-0.8557</td>\n",
       "      <td>.,1.445,.,1.445,.</td>\n",
       "      <td>-0.57,-0.9,-0.85,-0.9,-0.93</td>\n",
       "      <td>0.639596939087</td>\n",
       "      <td>0.212,0.212,0.212,0.212,0.212</td>\n",
       "      <td>0.487,0.487,0.499,0.486,0.499</td>\n",
       "      <td>5.97</td>\n",
       "      <td>16.4504</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>11:113412762-113412762</td>\n",
       "      <td>C</td>\n",
       "      <td>311</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>S/C</td>\n",
       "      <td>chr11:113412762</td>\n",
       "      <td>deleterious(0.05)</td>\n",
       "      <td>possibly_damaging(0.908)</td>\n",
       "      <td>-0.393485</td>\n",
       "      <td>-0.293969</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0033855736,0.0033855736,0.0033855736,0.00338...</td>\n",
       "      <td>-0.4235</td>\n",
       "      <td>.,2.505,.,2.505,.</td>\n",
       "      <td>-0.41,-0.6,-0.49,-0.6,-0.61</td>\n",
       "      <td>0.526918172836</td>\n",
       "      <td>0.230,0.230,0.230,0.230,0.230</td>\n",
       "      <td>0.32,0.244,0.263,0.244,0.183</td>\n",
       "      <td>5.73</td>\n",
       "      <td>19.9025</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>11:113412766-113412766</td>\n",
       "      <td>A</td>\n",
       "      <td>310</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>P/S</td>\n",
       "      <td>chr11:113412766</td>\n",
       "      <td>tolerated(0.61)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>-0.399698</td>\n",
       "      <td>-0.392466</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003302753,0.003302753,0.003302753,0.00330275...</td>\n",
       "      <td>-0.9782</td>\n",
       "      <td>.,1.18,.,1.18,.</td>\n",
       "      <td>-0.3,-0.19,-0.12,-0.19,-0.02</td>\n",
       "      <td>0.288126349449</td>\n",
       "      <td>0.053,0.053,0.053,0.053,0.053</td>\n",
       "      <td>0.095,0.081,0.085,0.08,0.084</td>\n",
       "      <td>1.56</td>\n",
       "      <td>7.7791</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>11:113416935-113416935</td>\n",
       "      <td>T</td>\n",
       "      <td>154</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>V/I</td>\n",
       "      <td>chr11:113416935</td>\n",
       "      <td>tolerated(0.16)</td>\n",
       "      <td>possibly_damaging(0.767)</td>\n",
       "      <td>0.134914</td>\n",
       "      <td>0.141922</td>\n",
       "      <td>...</td>\n",
       "      <td>0.35322282,0.35322282,0.35322282,0.35322282,0....</td>\n",
       "      <td>-0.3173</td>\n",
       "      <td>1.52,1.52,.,1.52,1.52,.</td>\n",
       "      <td>0.06,0.05,0.22,0.05,0.14,0.07</td>\n",
       "      <td>0.655525803566</td>\n",
       "      <td>0.324,0.324,0.324,0.324,0.324,0.324</td>\n",
       "      <td>0.517,0.619,0.635,0.624,0.685,.</td>\n",
       "      <td>5.51</td>\n",
       "      <td>19.4311</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>13:77901095-77901095</td>\n",
       "      <td>T</td>\n",
       "      <td>305</td>\n",
       "      <td>EDNRB</td>\n",
       "      <td>S/N</td>\n",
       "      <td>chr13:77901095</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0.006)</td>\n",
       "      <td>-0.601516</td>\n",
       "      <td>-0.625781</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0069636405,0.0069636405,0.0069636405,0.00696...</td>\n",
       "      <td>-1.0241</td>\n",
       "      <td>.,-1,-1,-1,-1,-1,-1,.,.</td>\n",
       "      <td>1.51,.,.,1.43,.,.,.,.,.</td>\n",
       "      <td>0.760299384594</td>\n",
       "      <td>0.095,.,.,0.095,.,.,.,.,.</td>\n",
       "      <td>0.073,.,.,0.085,.,0.093,.,.,.</td>\n",
       "      <td>5.69</td>\n",
       "      <td>13.0659</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64824</th>\n",
       "      <td>X:114907377-114907377</td>\n",
       "      <td>T</td>\n",
       "      <td>447</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>P/S</td>\n",
       "      <td>chrX:114907377</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.035)</td>\n",
       "      <td>-0.203906</td>\n",
       "      <td>-0.530674</td>\n",
       "      <td>...</td>\n",
       "      <td>0.08104396,0.08104396</td>\n",
       "      <td>-1.0440</td>\n",
       "      <td>.,.</td>\n",
       "      <td>0.6,0.6</td>\n",
       "      <td>0.311005502939</td>\n",
       "      <td>0.079,0.079</td>\n",
       "      <td>0.085,0.085</td>\n",
       "      <td>4.12</td>\n",
       "      <td>8.9218</td>\n",
       "      <td>0.183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64828</th>\n",
       "      <td>X:114907381-114907381</td>\n",
       "      <td>T</td>\n",
       "      <td>448</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>S/F</td>\n",
       "      <td>chrX:114907381</td>\n",
       "      <td>tolerated(0.2)</td>\n",
       "      <td>benign(0.194)</td>\n",
       "      <td>-0.218816</td>\n",
       "      <td>-0.55209</td>\n",
       "      <td>...</td>\n",
       "      <td>0.13516775,0.13516775</td>\n",
       "      <td>-0.9006</td>\n",
       "      <td>.,.</td>\n",
       "      <td>-1.63,-1.63</td>\n",
       "      <td>0.456193089485</td>\n",
       "      <td>0.028,0.028</td>\n",
       "      <td>0.094,0.174</td>\n",
       "      <td>3.16</td>\n",
       "      <td>12.2927</td>\n",
       "      <td>0.294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64832</th>\n",
       "      <td>X:114907387-114907387</td>\n",
       "      <td>C</td>\n",
       "      <td>450</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>V/A</td>\n",
       "      <td>chrX:114907387</td>\n",
       "      <td>tolerated(0.27)</td>\n",
       "      <td>benign(0.052)</td>\n",
       "      <td>-0.175516</td>\n",
       "      <td>-0.438056</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1508547,0.1508547</td>\n",
       "      <td>-0.9876</td>\n",
       "      <td>.,.</td>\n",
       "      <td>0.35,0.35</td>\n",
       "      <td>0.466780900955</td>\n",
       "      <td>0.161,0.161</td>\n",
       "      <td>0.254,0.257</td>\n",
       "      <td>5.05</td>\n",
       "      <td>11.6062</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64836</th>\n",
       "      <td>X:114907399-114907399</td>\n",
       "      <td>A</td>\n",
       "      <td>454</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>R/K</td>\n",
       "      <td>chrX:114907399</td>\n",
       "      <td>tolerated(0.67)</td>\n",
       "      <td>benign(0.048)</td>\n",
       "      <td>-0.281812</td>\n",
       "      <td>-0.590743</td>\n",
       "      <td>...</td>\n",
       "      <td>0.08362916,0.08362916</td>\n",
       "      <td>-1.0218</td>\n",
       "      <td>.,.</td>\n",
       "      <td>0.24,0.24</td>\n",
       "      <td>0.508506894112</td>\n",
       "      <td>0.076,0.076</td>\n",
       "      <td>0.182,0.143</td>\n",
       "      <td>4.11</td>\n",
       "      <td>9.0644</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64840</th>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>458</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>V/A</td>\n",
       "      <td>chrX:114907411</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>benign(0.309)</td>\n",
       "      <td>-0.00911951</td>\n",
       "      <td>-0.250876</td>\n",
       "      <td>...</td>\n",
       "      <td>0.33071065,0.33071065</td>\n",
       "      <td>-0.3392</td>\n",
       "      <td>.,.</td>\n",
       "      <td>-1.42,-1.42</td>\n",
       "      <td>0.682651281357</td>\n",
       "      <td>0.182,0.182</td>\n",
       "      <td>0.537,0.466</td>\n",
       "      <td>5.05</td>\n",
       "      <td>11.6062</td>\n",
       "      <td>0.22</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15580 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Location Allele Protein_position SYMBOL Amino_acids  \\\n",
       "1      11:113412715-113412715      C              327   DRD2         K/E   \n",
       "7      11:113412762-113412762      C              311   DRD2         S/C   \n",
       "13     11:113412766-113412766      A              310   DRD2         P/S   \n",
       "19     11:113416935-113416935      T              154   DRD2         V/I   \n",
       "29       13:77901095-77901095      T              305  EDNRB         S/N   \n",
       "...                       ...    ...              ...    ...         ...   \n",
       "64824   X:114907377-114907377      T              447  HTR2C         P/S   \n",
       "64828   X:114907381-114907381      T              448  HTR2C         S/F   \n",
       "64832   X:114907387-114907387      C              450  HTR2C         V/A   \n",
       "64836   X:114907399-114907399      A              454  HTR2C         R/K   \n",
       "64840   X:114907411-114907411      C              458  HTR2C         V/A   \n",
       "\n",
       "                 locus               SIFT                  PolyPhen  \\\n",
       "1      chr11:113412715    tolerated(0.27)             benign(0.031)   \n",
       "7      chr11:113412762  deleterious(0.05)  possibly_damaging(0.908)   \n",
       "13     chr11:113412766    tolerated(0.61)             benign(0.001)   \n",
       "19     chr11:113416935    tolerated(0.16)  possibly_damaging(0.767)   \n",
       "29      chr13:77901095    tolerated(0.92)             benign(0.006)   \n",
       "...                ...                ...                       ...   \n",
       "64824   chrX:114907377       tolerated(1)             benign(0.035)   \n",
       "64828   chrX:114907381     tolerated(0.2)             benign(0.194)   \n",
       "64832   chrX:114907387    tolerated(0.27)             benign(0.052)   \n",
       "64836   chrX:114907399    tolerated(0.67)             benign(0.048)   \n",
       "64840   chrX:114907411     deleterious(0)             benign(0.309)   \n",
       "\n",
       "      BayesDel_addAF_score BayesDel_noAF_score  ...  \\\n",
       "1                -0.260005           -0.140988  ...   \n",
       "7                -0.393485           -0.293969  ...   \n",
       "13               -0.399698           -0.392466  ...   \n",
       "19                0.134914            0.141922  ...   \n",
       "29               -0.601516           -0.625781  ...   \n",
       "...                    ...                 ...  ...   \n",
       "64824            -0.203906           -0.530674  ...   \n",
       "64828            -0.218816            -0.55209  ...   \n",
       "64832            -0.175516           -0.438056  ...   \n",
       "64836            -0.281812           -0.590743  ...   \n",
       "64840          -0.00911951           -0.250876  ...   \n",
       "\n",
       "                                           MetaRNN_score MetaSVM_score  \\\n",
       "1      0.006436944,0.006436944,0.006436944,0.00643694...       -0.8557   \n",
       "7      0.0033855736,0.0033855736,0.0033855736,0.00338...       -0.4235   \n",
       "13     0.003302753,0.003302753,0.003302753,0.00330275...       -0.9782   \n",
       "19     0.35322282,0.35322282,0.35322282,0.35322282,0....       -0.3173   \n",
       "29     0.0069636405,0.0069636405,0.0069636405,0.00696...       -1.0241   \n",
       "...                                                  ...           ...   \n",
       "64824                              0.08104396,0.08104396       -1.0440   \n",
       "64828                              0.13516775,0.13516775       -0.9006   \n",
       "64832                                0.1508547,0.1508547       -0.9876   \n",
       "64836                              0.08362916,0.08362916       -1.0218   \n",
       "64840                              0.33071065,0.33071065       -0.3392   \n",
       "\n",
       "        MutationAssessor_score                  PROVEAN_score PrimateAI_score  \\\n",
       "1            .,1.445,.,1.445,.    -0.57,-0.9,-0.85,-0.9,-0.93  0.639596939087   \n",
       "7            .,2.505,.,2.505,.    -0.41,-0.6,-0.49,-0.6,-0.61  0.526918172836   \n",
       "13             .,1.18,.,1.18,.   -0.3,-0.19,-0.12,-0.19,-0.02  0.288126349449   \n",
       "19     1.52,1.52,.,1.52,1.52,.  0.06,0.05,0.22,0.05,0.14,0.07  0.655525803566   \n",
       "29     .,-1,-1,-1,-1,-1,-1,.,.        1.51,.,.,1.43,.,.,.,.,.  0.760299384594   \n",
       "...                        ...                            ...             ...   \n",
       "64824                      .,.                        0.6,0.6  0.311005502939   \n",
       "64828                      .,.                    -1.63,-1.63  0.456193089485   \n",
       "64832                      .,.                      0.35,0.35  0.466780900955   \n",
       "64836                      .,.                      0.24,0.24  0.508506894112   \n",
       "64840                      .,.                    -1.42,-1.42  0.682651281357   \n",
       "\n",
       "                               REVEL_score                      VEST4_score  \\\n",
       "1            0.212,0.212,0.212,0.212,0.212    0.487,0.487,0.499,0.486,0.499   \n",
       "7            0.230,0.230,0.230,0.230,0.230     0.32,0.244,0.263,0.244,0.183   \n",
       "13           0.053,0.053,0.053,0.053,0.053     0.095,0.081,0.085,0.08,0.084   \n",
       "19     0.324,0.324,0.324,0.324,0.324,0.324  0.517,0.619,0.635,0.624,0.685,.   \n",
       "29               0.095,.,.,0.095,.,.,.,.,.    0.073,.,.,0.085,.,0.093,.,.,.   \n",
       "...                                    ...                              ...   \n",
       "64824                          0.079,0.079                      0.085,0.085   \n",
       "64828                          0.028,0.028                      0.094,0.174   \n",
       "64832                          0.161,0.161                      0.254,0.257   \n",
       "64836                          0.076,0.076                      0.182,0.143   \n",
       "64840                          0.182,0.182                      0.537,0.466   \n",
       "\n",
       "      GERP++_RS SiPhy_29way_logOdds MutPred_score  \n",
       "1          5.97             16.4504             -  \n",
       "7          5.73             19.9025             -  \n",
       "13         1.56              7.7791             -  \n",
       "19         5.51             19.4311             -  \n",
       "29         5.69             13.0659             -  \n",
       "...         ...                 ...           ...  \n",
       "64824      4.12              8.9218         0.183  \n",
       "64828      3.16             12.2927         0.294  \n",
       "64832      5.05             11.6062             -  \n",
       "64836      4.11              9.0644             -  \n",
       "64840      5.05             11.6062          0.22  \n",
       "\n",
       "[15580 rows x 31 columns]"
      ]
     },
     "execution_count": 348,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "extract_list = [\"Location\", 'Allele','Protein_position',\"SYMBOL\",'Amino_acids','locus']+VEP_list\n",
    "# extract corresponding vep columns from the original dataframe\n",
    "df2 = df1.loc[:,extract_list]\n",
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 335,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "68"
      ]
     },
     "execution_count": 335,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(np.unique(df2['SYMBOL']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 349,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>locus</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BayesDel_addAF_score</th>\n",
       "      <th>BayesDel_noAF_score</th>\n",
       "      <th>...</th>\n",
       "      <th>coding_description</th>\n",
       "      <th>category</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>AF.Cases</th>\n",
       "      <th>AF.Controls</th>\n",
       "      <th>Pvalue</th>\n",
       "      <th>vcf</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>327</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>K/E</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>tolerated(0.27)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td>-0.260005</td>\n",
       "      <td>-0.140988</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>1907</td>\n",
       "      <td>0.002415</td>\n",
       "      <td>0.431260</td>\n",
       "      <td>0.53696</td>\n",
       "      <td>0.003472</td>\n",
       "      <td>0.002413</td>\n",
       "      <td>0.421890</td>\n",
       "      <td>11 113412715 . T C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>327</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>K/E</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>tolerated(0.27)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td>-0.260005</td>\n",
       "      <td>-0.140988</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>1907</td>\n",
       "      <td>0.002415</td>\n",
       "      <td>-0.246980</td>\n",
       "      <td>0.61625</td>\n",
       "      <td>0.001845</td>\n",
       "      <td>0.002416</td>\n",
       "      <td>0.688580</td>\n",
       "      <td>11 113412715 . T C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>327</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>K/E</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>tolerated(0.27)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td>-0.260005</td>\n",
       "      <td>-0.140988</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>1907</td>\n",
       "      <td>0.002415</td>\n",
       "      <td>1.305800</td>\n",
       "      <td>1.07300</td>\n",
       "      <td>0.005556</td>\n",
       "      <td>0.002413</td>\n",
       "      <td>0.223580</td>\n",
       "      <td>11 113412715 . T C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>327</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>K/E</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>tolerated(0.27)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td>-0.260005</td>\n",
       "      <td>-0.140988</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>1907</td>\n",
       "      <td>0.002415</td>\n",
       "      <td>-0.141640</td>\n",
       "      <td>0.27161</td>\n",
       "      <td>0.002068</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>0.602020</td>\n",
       "      <td>11 113412715 . T C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>327</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>K/E</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>tolerated(0.27)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td>-0.260005</td>\n",
       "      <td>-0.140988</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>1907</td>\n",
       "      <td>0.002415</td>\n",
       "      <td>0.071904</td>\n",
       "      <td>0.52037</td>\n",
       "      <td>0.002584</td>\n",
       "      <td>0.002415</td>\n",
       "      <td>0.890100</td>\n",
       "      <td>11 113412715 . T C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>654733</th>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>458</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>V/A</td>\n",
       "      <td>chrX:114907411</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>benign(0.309)</td>\n",
       "      <td>-0.00911951</td>\n",
       "      <td>-0.250876</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter XXI Factors influencing health status ...</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>-1.007700</td>\n",
       "      <td>7.82990</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.897600</td>\n",
       "      <td>X 114907411 . T C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>654734</th>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>458</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>V/A</td>\n",
       "      <td>chrX:114907411</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>benign(0.309)</td>\n",
       "      <td>-0.00911951</td>\n",
       "      <td>-0.250876</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter XXI Factors influencing health status ...</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>5.129800</td>\n",
       "      <td>2.33040</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.027721</td>\n",
       "      <td>X 114907411 . T C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>654735</th>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>458</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>V/A</td>\n",
       "      <td>chrX:114907411</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>benign(0.309)</td>\n",
       "      <td>-0.00911951</td>\n",
       "      <td>-0.250876</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter XXI Factors influencing health status ...</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>-1.015900</td>\n",
       "      <td>4.91580</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.836280</td>\n",
       "      <td>X 114907411 . T C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>654736</th>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>458</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>V/A</td>\n",
       "      <td>chrX:114907411</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>benign(0.309)</td>\n",
       "      <td>-0.00911951</td>\n",
       "      <td>-0.250876</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter XXI Factors influencing health status ...</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>-1.022800</td>\n",
       "      <td>3.18950</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.748460</td>\n",
       "      <td>X 114907411 . T C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>654737</th>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>458</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>V/A</td>\n",
       "      <td>chrX:114907411</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>benign(0.309)</td>\n",
       "      <td>-0.00911951</td>\n",
       "      <td>-0.250876</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter XXI Factors influencing health status ...</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>-1.054300</td>\n",
       "      <td>2.81690</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.708210</td>\n",
       "      <td>X 114907411 . T C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>654738 rows × 63 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Location Allele Protein_position SYMBOL Amino_acids  \\\n",
       "0       11:113412715-113412715      C              327   DRD2         K/E   \n",
       "1       11:113412715-113412715      C              327   DRD2         K/E   \n",
       "2       11:113412715-113412715      C              327   DRD2         K/E   \n",
       "3       11:113412715-113412715      C              327   DRD2         K/E   \n",
       "4       11:113412715-113412715      C              327   DRD2         K/E   \n",
       "...                        ...    ...              ...    ...         ...   \n",
       "654733   X:114907411-114907411      C              458  HTR2C         V/A   \n",
       "654734   X:114907411-114907411      C              458  HTR2C         V/A   \n",
       "654735   X:114907411-114907411      C              458  HTR2C         V/A   \n",
       "654736   X:114907411-114907411      C              458  HTR2C         V/A   \n",
       "654737   X:114907411-114907411      C              458  HTR2C         V/A   \n",
       "\n",
       "                  locus             SIFT       PolyPhen BayesDel_addAF_score  \\\n",
       "0       chr11:113412715  tolerated(0.27)  benign(0.031)            -0.260005   \n",
       "1       chr11:113412715  tolerated(0.27)  benign(0.031)            -0.260005   \n",
       "2       chr11:113412715  tolerated(0.27)  benign(0.031)            -0.260005   \n",
       "3       chr11:113412715  tolerated(0.27)  benign(0.031)            -0.260005   \n",
       "4       chr11:113412715  tolerated(0.27)  benign(0.031)            -0.260005   \n",
       "...                 ...              ...            ...                  ...   \n",
       "654733   chrX:114907411   deleterious(0)  benign(0.309)          -0.00911951   \n",
       "654734   chrX:114907411   deleterious(0)  benign(0.309)          -0.00911951   \n",
       "654735   chrX:114907411   deleterious(0)  benign(0.309)          -0.00911951   \n",
       "654736   chrX:114907411   deleterious(0)  benign(0.309)          -0.00911951   \n",
       "654737   chrX:114907411   deleterious(0)  benign(0.309)          -0.00911951   \n",
       "\n",
       "       BayesDel_noAF_score  ... coding_description  \\\n",
       "0                -0.140988  ...                NaN   \n",
       "1                -0.140988  ...                NaN   \n",
       "2                -0.140988  ...                NaN   \n",
       "3                -0.140988  ...                NaN   \n",
       "4                -0.140988  ...                NaN   \n",
       "...                    ...  ...                ...   \n",
       "654733           -0.250876  ...                NaN   \n",
       "654734           -0.250876  ...                NaN   \n",
       "654735           -0.250876  ...                NaN   \n",
       "654736           -0.250876  ...                NaN   \n",
       "654737           -0.250876  ...                NaN   \n",
       "\n",
       "                                                 category    AC        AF  \\\n",
       "0       Chapter II Neoplasms | C15-C26 Malignant neopl...  1907  0.002415   \n",
       "1       Chapter II Neoplasms | C15-C26 Malignant neopl...  1907  0.002415   \n",
       "2       Chapter II Neoplasms | C15-C26 Malignant neopl...  1907  0.002415   \n",
       "3       Chapter II Neoplasms | C15-C26 Malignant neopl...  1907  0.002415   \n",
       "4       Chapter II Neoplasms | C15-C26 Malignant neopl...  1907  0.002415   \n",
       "...                                                   ...   ...       ...   \n",
       "654733  Chapter XXI Factors influencing health status ...     3  0.000004   \n",
       "654734  Chapter XXI Factors influencing health status ...     3  0.000004   \n",
       "654735  Chapter XXI Factors influencing health status ...     3  0.000004   \n",
       "654736  Chapter XXI Factors influencing health status ...     3  0.000004   \n",
       "654737  Chapter XXI Factors influencing health status ...     3  0.000004   \n",
       "\n",
       "            BETA       SE  AF.Cases AF.Controls    Pvalue                 vcf  \n",
       "0       0.431260  0.53696  0.003472    0.002413  0.421890  11 113412715 . T C  \n",
       "1      -0.246980  0.61625  0.001845    0.002416  0.688580  11 113412715 . T C  \n",
       "2       1.305800  1.07300  0.005556    0.002413  0.223580  11 113412715 . T C  \n",
       "3      -0.141640  0.27161  0.002068    0.002417  0.602020  11 113412715 . T C  \n",
       "4       0.071904  0.52037  0.002584    0.002415  0.890100  11 113412715 . T C  \n",
       "...          ...      ...       ...         ...       ...                 ...  \n",
       "654733 -1.007700  7.82990  0.000000    0.000004  0.897600   X 114907411 . T C  \n",
       "654734  5.129800  2.33040  0.000049    0.000003  0.027721   X 114907411 . T C  \n",
       "654735 -1.015900  4.91580  0.000000    0.000004  0.836280   X 114907411 . T C  \n",
       "654736 -1.022800  3.18950  0.000000    0.000004  0.748460   X 114907411 . T C  \n",
       "654737 -1.054300  2.81690  0.000000    0.000004  0.708210   X 114907411 . T C  \n",
       "\n",
       "[654738 rows x 63 columns]"
      ]
     },
     "execution_count": 349,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# merge two\n",
    "VEPwithGB_icd10 = pd.merge(df2,icd10_df,on=['locus','Allele'])\n",
    "VEPwithGB_icd10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 286,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "68"
      ]
     },
     "execution_count": 286,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(np.unique(VEPwithGB_icd10['SYMBOL']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "metadata": {},
   "outputs": [],
   "source": [
    "VEPwithGB_icd10.to_csv('VEPwithGB_icd10_0323.csv',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 351,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>locus</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BayesDel_addAF_score</th>\n",
       "      <th>BayesDel_noAF_score</th>\n",
       "      <th>...</th>\n",
       "      <th>coding_description</th>\n",
       "      <th>category</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>AF.Cases</th>\n",
       "      <th>AF.Controls</th>\n",
       "      <th>Pvalue</th>\n",
       "      <th>vcf</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>327</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>K/E</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>tolerated(0.27)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td>-0.260005</td>\n",
       "      <td>-0.140988</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>1907</td>\n",
       "      <td>0.002415</td>\n",
       "      <td>0.43126</td>\n",
       "      <td>0.53696</td>\n",
       "      <td>3.472200e-03</td>\n",
       "      <td>0.002413</td>\n",
       "      <td>0.421890</td>\n",
       "      <td>11 113412715 . T C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>11:113412762-113412762</td>\n",
       "      <td>C</td>\n",
       "      <td>311</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>S/C</td>\n",
       "      <td>chr11:113412762</td>\n",
       "      <td>deleterious(0.05)</td>\n",
       "      <td>possibly_damaging(0.908)</td>\n",
       "      <td>-0.393485</td>\n",
       "      <td>-0.293969</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>16012</td>\n",
       "      <td>0.020277</td>\n",
       "      <td>-0.14869</td>\n",
       "      <td>0.18710</td>\n",
       "      <td>1.736100e-02</td>\n",
       "      <td>0.020282</td>\n",
       "      <td>0.426780</td>\n",
       "      <td>11 113412762 . G C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>11:113412766-113412766</td>\n",
       "      <td>A</td>\n",
       "      <td>310</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>P/S</td>\n",
       "      <td>chr11:113412766</td>\n",
       "      <td>tolerated(0.61)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>-0.399698</td>\n",
       "      <td>-0.392466</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>235</td>\n",
       "      <td>0.000298</td>\n",
       "      <td>-1.00910</td>\n",
       "      <td>1.63230</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000298</td>\n",
       "      <td>0.536450</td>\n",
       "      <td>11 113412766 . G A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>11:113416935-113416935</td>\n",
       "      <td>T</td>\n",
       "      <td>154</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>V/I</td>\n",
       "      <td>chr11:113416935</td>\n",
       "      <td>tolerated(0.16)</td>\n",
       "      <td>possibly_damaging(0.767)</td>\n",
       "      <td>0.134914</td>\n",
       "      <td>0.141922</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>12</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>-1.00480</td>\n",
       "      <td>6.30530</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>0.873390</td>\n",
       "      <td>11 113416935 . C T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>13:77901095-77901095</td>\n",
       "      <td>T</td>\n",
       "      <td>305</td>\n",
       "      <td>EDNRB</td>\n",
       "      <td>S/N</td>\n",
       "      <td>chr13:77901095</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0.006)</td>\n",
       "      <td>-0.601516</td>\n",
       "      <td>-0.625781</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>11381</td>\n",
       "      <td>0.014412</td>\n",
       "      <td>-0.42015</td>\n",
       "      <td>0.22281</td>\n",
       "      <td>8.333300e-03</td>\n",
       "      <td>0.014423</td>\n",
       "      <td>0.059333</td>\n",
       "      <td>13 77901095 . C T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>654517</th>\n",
       "      <td>X:114907377-114907377</td>\n",
       "      <td>T</td>\n",
       "      <td>447</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>P/S</td>\n",
       "      <td>chrX:114907377</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.035)</td>\n",
       "      <td>-0.203906</td>\n",
       "      <td>-0.530674</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>-0.56465</td>\n",
       "      <td>9.73350</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.953740</td>\n",
       "      <td>X 114907377 . C T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>654562</th>\n",
       "      <td>X:114907381-114907381</td>\n",
       "      <td>T</td>\n",
       "      <td>448</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>S/F</td>\n",
       "      <td>chrX:114907381</td>\n",
       "      <td>tolerated(0.2)</td>\n",
       "      <td>benign(0.194)</td>\n",
       "      <td>-0.218816</td>\n",
       "      <td>-0.55209</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>-1.00070</td>\n",
       "      <td>19.87100</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.959840</td>\n",
       "      <td>X 114907381 . C T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>654605</th>\n",
       "      <td>X:114907387-114907387</td>\n",
       "      <td>C</td>\n",
       "      <td>450</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>V/A</td>\n",
       "      <td>chrX:114907387</td>\n",
       "      <td>tolerated(0.27)</td>\n",
       "      <td>benign(0.052)</td>\n",
       "      <td>-0.175516</td>\n",
       "      <td>-0.438056</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>12</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>-0.63211</td>\n",
       "      <td>6.53650</td>\n",
       "      <td>4.941800e-07</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>0.922960</td>\n",
       "      <td>X 114907387 . T C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>654650</th>\n",
       "      <td>X:114907399-114907399</td>\n",
       "      <td>A</td>\n",
       "      <td>454</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>R/K</td>\n",
       "      <td>chrX:114907399</td>\n",
       "      <td>tolerated(0.67)</td>\n",
       "      <td>benign(0.048)</td>\n",
       "      <td>-0.281812</td>\n",
       "      <td>-0.590743</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>35</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>-0.55746</td>\n",
       "      <td>2.88220</td>\n",
       "      <td>3.568200e-06</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0.846640</td>\n",
       "      <td>X 114907399 . G A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>654695</th>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>458</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>V/A</td>\n",
       "      <td>chrX:114907411</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>benign(0.309)</td>\n",
       "      <td>-0.00911951</td>\n",
       "      <td>-0.250876</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chapter II Neoplasms | C15-C26 Malignant neopl...</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>-1.00080</td>\n",
       "      <td>25.76100</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.969010</td>\n",
       "      <td>X 114907411 . T C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15580 rows × 63 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Location Allele Protein_position SYMBOL Amino_acids  \\\n",
       "0       11:113412715-113412715      C              327   DRD2         K/E   \n",
       "45      11:113412762-113412762      C              311   DRD2         S/C   \n",
       "90      11:113412766-113412766      A              310   DRD2         P/S   \n",
       "135     11:113416935-113416935      T              154   DRD2         V/I   \n",
       "180       13:77901095-77901095      T              305  EDNRB         S/N   \n",
       "...                        ...    ...              ...    ...         ...   \n",
       "654517   X:114907377-114907377      T              447  HTR2C         P/S   \n",
       "654562   X:114907381-114907381      T              448  HTR2C         S/F   \n",
       "654605   X:114907387-114907387      C              450  HTR2C         V/A   \n",
       "654650   X:114907399-114907399      A              454  HTR2C         R/K   \n",
       "654695   X:114907411-114907411      C              458  HTR2C         V/A   \n",
       "\n",
       "                  locus               SIFT                  PolyPhen  \\\n",
       "0       chr11:113412715    tolerated(0.27)             benign(0.031)   \n",
       "45      chr11:113412762  deleterious(0.05)  possibly_damaging(0.908)   \n",
       "90      chr11:113412766    tolerated(0.61)             benign(0.001)   \n",
       "135     chr11:113416935    tolerated(0.16)  possibly_damaging(0.767)   \n",
       "180      chr13:77901095    tolerated(0.92)             benign(0.006)   \n",
       "...                 ...                ...                       ...   \n",
       "654517   chrX:114907377       tolerated(1)             benign(0.035)   \n",
       "654562   chrX:114907381     tolerated(0.2)             benign(0.194)   \n",
       "654605   chrX:114907387    tolerated(0.27)             benign(0.052)   \n",
       "654650   chrX:114907399    tolerated(0.67)             benign(0.048)   \n",
       "654695   chrX:114907411     deleterious(0)             benign(0.309)   \n",
       "\n",
       "       BayesDel_addAF_score BayesDel_noAF_score  ... coding_description  \\\n",
       "0                 -0.260005           -0.140988  ...                NaN   \n",
       "45                -0.393485           -0.293969  ...                NaN   \n",
       "90                -0.399698           -0.392466  ...                NaN   \n",
       "135                0.134914            0.141922  ...                NaN   \n",
       "180               -0.601516           -0.625781  ...                NaN   \n",
       "...                     ...                 ...  ...                ...   \n",
       "654517            -0.203906           -0.530674  ...                NaN   \n",
       "654562            -0.218816            -0.55209  ...                NaN   \n",
       "654605            -0.175516           -0.438056  ...                NaN   \n",
       "654650            -0.281812           -0.590743  ...                NaN   \n",
       "654695          -0.00911951           -0.250876  ...                NaN   \n",
       "\n",
       "                                                 category     AC        AF  \\\n",
       "0       Chapter II Neoplasms | C15-C26 Malignant neopl...   1907  0.002415   \n",
       "45      Chapter II Neoplasms | C15-C26 Malignant neopl...  16012  0.020277   \n",
       "90      Chapter II Neoplasms | C15-C26 Malignant neopl...    235  0.000298   \n",
       "135     Chapter II Neoplasms | C15-C26 Malignant neopl...     12  0.000015   \n",
       "180     Chapter II Neoplasms | C15-C26 Malignant neopl...  11381  0.014412   \n",
       "...                                                   ...    ...       ...   \n",
       "654517  Chapter II Neoplasms | C15-C26 Malignant neopl...      4  0.000005   \n",
       "654562  Chapter II Neoplasms | C15-C26 Malignant neopl...      5  0.000006   \n",
       "654605  Chapter II Neoplasms | C15-C26 Malignant neopl...     12  0.000015   \n",
       "654650  Chapter II Neoplasms | C15-C26 Malignant neopl...     35  0.000045   \n",
       "654695  Chapter II Neoplasms | C15-C26 Malignant neopl...      3  0.000004   \n",
       "\n",
       "           BETA        SE      AF.Cases AF.Controls    Pvalue  \\\n",
       "0       0.43126   0.53696  3.472200e-03    0.002413  0.421890   \n",
       "45     -0.14869   0.18710  1.736100e-02    0.020282  0.426780   \n",
       "90     -1.00910   1.63230  0.000000e+00    0.000298  0.536450   \n",
       "135    -1.00480   6.30530  0.000000e+00    0.000015  0.873390   \n",
       "180    -0.42015   0.22281  8.333300e-03    0.014423  0.059333   \n",
       "...         ...       ...           ...         ...       ...   \n",
       "654517 -0.56465   9.73350  0.000000e+00    0.000005  0.953740   \n",
       "654562 -1.00070  19.87100  0.000000e+00    0.000006  0.959840   \n",
       "654605 -0.63211   6.53650  4.941800e-07    0.000015  0.922960   \n",
       "654650 -0.55746   2.88220  3.568200e-06    0.000045  0.846640   \n",
       "654695 -1.00080  25.76100  0.000000e+00    0.000004  0.969010   \n",
       "\n",
       "                       vcf  \n",
       "0       11 113412715 . T C  \n",
       "45      11 113412762 . G C  \n",
       "90      11 113412766 . G A  \n",
       "135     11 113416935 . C T  \n",
       "180      13 77901095 . C T  \n",
       "...                    ...  \n",
       "654517   X 114907377 . C T  \n",
       "654562   X 114907381 . C T  \n",
       "654605   X 114907387 . T C  \n",
       "654650   X 114907399 . G A  \n",
       "654695   X 114907411 . T C  \n",
       "\n",
       "[15580 rows x 63 columns]"
      ]
     },
     "execution_count": 351,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop_df = VEPwithGB_icd10.drop_duplicates(subset=['SYMBOL','Protein_position','Amino_acids'],keep='first')\n",
    "drop_df = VEPwithGB_icd10.drop_duplicates(subset=['locus','Allele'],keep='first')\n",
    "drop_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Completeness:\n",
    "def completeness(VEP_list,df):\n",
    "    stat_df = pd.DataFrame()\n",
    "    row = 0\n",
    "    for i in range(len(VEP_list)): \n",
    "        vep = VEP_list[i]\n",
    "        count_vep_pred = df[df[vep]!='-'].shape[0]\n",
    "        stat_df.loc[row,'vep'] = vep\n",
    "        # 21908 is the number of mutations I put into Emsembl\n",
    "        stat_df.loc[row,'Variants with prediction'] = str(count_vep_pred)+'/'+str(df.shape[0])\n",
    "        stat_df.loc[row,'Variants with prediction(%)'] = str(\"{:.2f}\".format(count_vep_pred/df.shape[0]*100))+\"%\"\n",
    "        row = row + 1\n",
    "\n",
    "    return stat_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Since for genebass analysis, the inputs are nucleotide variants, in order to calculate the completeness of different VEP predictions,\n",
    "# for VEP from Ensembl, we calculate the proportion of valid predictions over all nucleotide variants:\n",
    "# proportion for each VEP = number of valid predictions (for nucleotide variants) /all nucleotide variants\n",
    "# for Other VEPs outside Ensembl: Dynamut2 and VESPA, there inputs are amino acid variants, so their proportions are calculated:\n",
    "# proportion for each VEP = number of valid predictions (for amino acid variants) /all amino acid variants\n",
    "stat = completeness(VEP_list,drop_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 353,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>vep</th>\n",
       "      <th>Variants with prediction</th>\n",
       "      <th>Variants with prediction(%)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SIFT</td>\n",
       "      <td>15580/15580</td>\n",
       "      <td>100.00%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PolyPhen</td>\n",
       "      <td>15580/15580</td>\n",
       "      <td>100.00%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BayesDel_addAF_score</td>\n",
       "      <td>15578/15580</td>\n",
       "      <td>99.99%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BayesDel_noAF_score</td>\n",
       "      <td>15578/15580</td>\n",
       "      <td>99.99%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CADD_phred</td>\n",
       "      <td>15580/15580</td>\n",
       "      <td>100.00%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ClinPred_score</td>\n",
       "      <td>15578/15580</td>\n",
       "      <td>99.99%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>DANN_score</td>\n",
       "      <td>15578/15580</td>\n",
       "      <td>99.99%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>DEOGEN2_score</td>\n",
       "      <td>14488/15580</td>\n",
       "      <td>92.99%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>FATHMM_score</td>\n",
       "      <td>15327/15580</td>\n",
       "      <td>98.38%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>integrated_fitCons_score</td>\n",
       "      <td>15400/15580</td>\n",
       "      <td>98.84%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>LIST-S2_score</td>\n",
       "      <td>14564/15580</td>\n",
       "      <td>93.48%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>LRT_score</td>\n",
       "      <td>14000/15580</td>\n",
       "      <td>89.86%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>M-CAP_score</td>\n",
       "      <td>15366/15580</td>\n",
       "      <td>98.63%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MVP_score</td>\n",
       "      <td>15550/15580</td>\n",
       "      <td>99.81%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MetaLR_score</td>\n",
       "      <td>15578/15580</td>\n",
       "      <td>99.99%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MetaRNN_score</td>\n",
       "      <td>15580/15580</td>\n",
       "      <td>100.00%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MetaSVM_score</td>\n",
       "      <td>15578/15580</td>\n",
       "      <td>99.99%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>MutationAssessor_score</td>\n",
       "      <td>14473/15580</td>\n",
       "      <td>92.89%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PROVEAN_score</td>\n",
       "      <td>15327/15580</td>\n",
       "      <td>98.38%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PrimateAI_score</td>\n",
       "      <td>15499/15580</td>\n",
       "      <td>99.48%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>REVEL_score</td>\n",
       "      <td>15327/15580</td>\n",
       "      <td>98.38%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>VEST4_score</td>\n",
       "      <td>15580/15580</td>\n",
       "      <td>100.00%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GERP++_RS</td>\n",
       "      <td>15575/15580</td>\n",
       "      <td>99.97%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SiPhy_29way_logOdds</td>\n",
       "      <td>15574/15580</td>\n",
       "      <td>99.96%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>MutPred_score</td>\n",
       "      <td>12956/15580</td>\n",
       "      <td>83.16%</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         vep Variants with prediction  \\\n",
       "0                       SIFT              15580/15580   \n",
       "1                   PolyPhen              15580/15580   \n",
       "2       BayesDel_addAF_score              15578/15580   \n",
       "3        BayesDel_noAF_score              15578/15580   \n",
       "4                 CADD_phred              15580/15580   \n",
       "5             ClinPred_score              15578/15580   \n",
       "6                 DANN_score              15578/15580   \n",
       "7              DEOGEN2_score              14488/15580   \n",
       "8               FATHMM_score              15327/15580   \n",
       "9   integrated_fitCons_score              15400/15580   \n",
       "10             LIST-S2_score              14564/15580   \n",
       "11                 LRT_score              14000/15580   \n",
       "12               M-CAP_score              15366/15580   \n",
       "13                 MVP_score              15550/15580   \n",
       "14              MetaLR_score              15578/15580   \n",
       "15             MetaRNN_score              15580/15580   \n",
       "16             MetaSVM_score              15578/15580   \n",
       "17    MutationAssessor_score              14473/15580   \n",
       "18             PROVEAN_score              15327/15580   \n",
       "19           PrimateAI_score              15499/15580   \n",
       "20               REVEL_score              15327/15580   \n",
       "21               VEST4_score              15580/15580   \n",
       "22                 GERP++_RS              15575/15580   \n",
       "23       SiPhy_29way_logOdds              15574/15580   \n",
       "24             MutPred_score              12956/15580   \n",
       "\n",
       "   Variants with prediction(%)  \n",
       "0                      100.00%  \n",
       "1                      100.00%  \n",
       "2                       99.99%  \n",
       "3                       99.99%  \n",
       "4                      100.00%  \n",
       "5                       99.99%  \n",
       "6                       99.99%  \n",
       "7                       92.99%  \n",
       "8                       98.38%  \n",
       "9                       98.84%  \n",
       "10                      93.48%  \n",
       "11                      89.86%  \n",
       "12                      98.63%  \n",
       "13                      99.81%  \n",
       "14                      99.99%  \n",
       "15                     100.00%  \n",
       "16                      99.99%  \n",
       "17                      92.89%  \n",
       "18                      98.38%  \n",
       "19                      99.48%  \n",
       "20                      98.38%  \n",
       "21                     100.00%  \n",
       "22                      99.97%  \n",
       "23                      99.96%  \n",
       "24                      83.16%  "
      ]
     },
     "execution_count": 353,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 354,
   "metadata": {},
   "outputs": [],
   "source": [
    "stat.to_csv('Genebass_completeness_new.csv',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "773d181160c386017ad51b585d282f9f41a012296bacf0d61cb3b8d5443ae20a"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
