{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "from sklearn.metrics import r2_score\n",
    "import re\n",
    "import requests\n",
    "import json\n",
    "import os\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>ENST</th>\n",
       "      <th>ENSG</th>\n",
       "      <th>protein</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Uniprot</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>ENST00000646641</td>\n",
       "      <td>ENSG00000267534</td>\n",
       "      <td>s1pr2_human</td>\n",
       "      <td>S1PR2</td>\n",
       "      <td>O95136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>121</td>\n",
       "      <td>ENST00000547270</td>\n",
       "      <td>ENSG00000257138</td>\n",
       "      <td>t2r38_human</td>\n",
       "      <td>TAS2R38</td>\n",
       "      <td>P59533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>219</td>\n",
       "      <td>ENST00000390675</td>\n",
       "      <td>ENSG00000256436</td>\n",
       "      <td>t2r31_human</td>\n",
       "      <td>TAS2R31</td>\n",
       "      <td>P59538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>344</td>\n",
       "      <td>ENST00000539585</td>\n",
       "      <td>ENSG00000256188</td>\n",
       "      <td>t2r30_human</td>\n",
       "      <td>TAS2R30</td>\n",
       "      <td>P59541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>483</td>\n",
       "      <td>ENST00000538986</td>\n",
       "      <td>ENSG00000255837</td>\n",
       "      <td>t2r20_human</td>\n",
       "      <td>TAS2R20</td>\n",
       "      <td>P59543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>64029</td>\n",
       "      <td>ENST00000510937</td>\n",
       "      <td>ENSG00000226306</td>\n",
       "      <td>npy6r_human</td>\n",
       "      <td>NPY6R</td>\n",
       "      <td>Q99463</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>64030</td>\n",
       "      <td>ENST00000641193</td>\n",
       "      <td>ENSG00000279301</td>\n",
       "      <td>o2t11_human</td>\n",
       "      <td>OR2T11</td>\n",
       "      <td>Q8NH01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>397</th>\n",
       "      <td>64031</td>\n",
       "      <td>ENST00000641732</td>\n",
       "      <td>ENSG00000172146</td>\n",
       "      <td>or1a1_human</td>\n",
       "      <td>OR1A1</td>\n",
       "      <td>Q9P1Q5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>398</th>\n",
       "      <td>64032</td>\n",
       "      <td>ENST00000328890</td>\n",
       "      <td>ENSG00000183024</td>\n",
       "      <td>or1g1_human</td>\n",
       "      <td>OR1G1</td>\n",
       "      <td>P47890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399</th>\n",
       "      <td>64034</td>\n",
       "      <td>ENST00000466706</td>\n",
       "      <td>ENSG00000179073</td>\n",
       "      <td>taar3_human</td>\n",
       "      <td>TAAR3P</td>\n",
       "      <td>Q9P1P4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>400 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     index             ENST             ENSG      protein   SYMBOL Uniprot\n",
       "0        0  ENST00000646641  ENSG00000267534  s1pr2_human    S1PR2  O95136\n",
       "1      121  ENST00000547270  ENSG00000257138  t2r38_human  TAS2R38  P59533\n",
       "2      219  ENST00000390675  ENSG00000256436  t2r31_human  TAS2R31  P59538\n",
       "3      344  ENST00000539585  ENSG00000256188  t2r30_human  TAS2R30  P59541\n",
       "4      483  ENST00000538986  ENSG00000255837  t2r20_human  TAS2R20  P59543\n",
       "..     ...              ...              ...          ...      ...     ...\n",
       "395  64029  ENST00000510937  ENSG00000226306  npy6r_human    NPY6R  Q99463\n",
       "396  64030  ENST00000641193  ENSG00000279301  o2t11_human   OR2T11  Q8NH01\n",
       "397  64031  ENST00000641732  ENSG00000172146  or1a1_human    OR1A1  Q9P1Q5\n",
       "398  64032  ENST00000328890  ENSG00000183024  or1g1_human    OR1G1  P47890\n",
       "399  64034  ENST00000466706  ENSG00000179073  taar3_human   TAAR3P  Q9P1P4\n",
       "\n",
       "[400 rows x 6 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in all gene names\n",
    "gene_list_raw =pd.read_csv(\"230323_EST_ENSG_GENE_new.csv\")\n",
    "gene_list_raw = gene_list_raw.rename(columns={'gene':'SYMBOL'})\n",
    "gene_list_raw"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/joannahench/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3444: DtypeWarning: Columns (8,9,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>locus</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BayesDel_addAF_score</th>\n",
       "      <th>...</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>AF.Cases</th>\n",
       "      <th>AF.Controls</th>\n",
       "      <th>Pvalue</th>\n",
       "      <th>vcf</th>\n",
       "      <th>mutation_from</th>\n",
       "      <th>mutation_to</th>\n",
       "      <th>protein</th>\n",
       "      <th>Uniprot</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>11:113412715-113412715</td>\n",
       "      <td>C</td>\n",
       "      <td>327</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>K/E</td>\n",
       "      <td>chr11:113412715</td>\n",
       "      <td>tolerated(0.27)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td>-0.260005</td>\n",
       "      <td>...</td>\n",
       "      <td>0.43126</td>\n",
       "      <td>0.53696</td>\n",
       "      <td>3.472200e-03</td>\n",
       "      <td>0.002413</td>\n",
       "      <td>0.421890</td>\n",
       "      <td>11 113412715 . T C</td>\n",
       "      <td>K</td>\n",
       "      <td>E</td>\n",
       "      <td>drd2_human</td>\n",
       "      <td>P14416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>45</td>\n",
       "      <td>11:113412762-113412762</td>\n",
       "      <td>C</td>\n",
       "      <td>311</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>S/C</td>\n",
       "      <td>chr11:113412762</td>\n",
       "      <td>deleterious(0.05)</td>\n",
       "      <td>possibly_damaging(0.908)</td>\n",
       "      <td>-0.393485</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.14869</td>\n",
       "      <td>0.18710</td>\n",
       "      <td>1.736100e-02</td>\n",
       "      <td>0.020282</td>\n",
       "      <td>0.426780</td>\n",
       "      <td>11 113412762 . G C</td>\n",
       "      <td>S</td>\n",
       "      <td>C</td>\n",
       "      <td>drd2_human</td>\n",
       "      <td>P14416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>90</td>\n",
       "      <td>11:113412766-113412766</td>\n",
       "      <td>A</td>\n",
       "      <td>310</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>P/S</td>\n",
       "      <td>chr11:113412766</td>\n",
       "      <td>tolerated(0.61)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>-0.399698</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.00910</td>\n",
       "      <td>1.63230</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000298</td>\n",
       "      <td>0.536450</td>\n",
       "      <td>11 113412766 . G A</td>\n",
       "      <td>P</td>\n",
       "      <td>S</td>\n",
       "      <td>drd2_human</td>\n",
       "      <td>P14416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>135</td>\n",
       "      <td>11:113416935-113416935</td>\n",
       "      <td>T</td>\n",
       "      <td>154</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>V/I</td>\n",
       "      <td>chr11:113416935</td>\n",
       "      <td>tolerated(0.16)</td>\n",
       "      <td>possibly_damaging(0.767)</td>\n",
       "      <td>0.134914</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.00480</td>\n",
       "      <td>6.30530</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>0.873390</td>\n",
       "      <td>11 113416935 . C T</td>\n",
       "      <td>V</td>\n",
       "      <td>I</td>\n",
       "      <td>drd2_human</td>\n",
       "      <td>P14416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>180</td>\n",
       "      <td>13:77901095-77901095</td>\n",
       "      <td>T</td>\n",
       "      <td>305</td>\n",
       "      <td>EDNRB</td>\n",
       "      <td>S/N</td>\n",
       "      <td>chr13:77901095</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0.006)</td>\n",
       "      <td>-0.601516</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.42015</td>\n",
       "      <td>0.22281</td>\n",
       "      <td>8.333300e-03</td>\n",
       "      <td>0.014423</td>\n",
       "      <td>0.059333</td>\n",
       "      <td>13 77901095 . C T</td>\n",
       "      <td>S</td>\n",
       "      <td>N</td>\n",
       "      <td>ednrb_human</td>\n",
       "      <td>P24530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15486</th>\n",
       "      <td>654517</td>\n",
       "      <td>X:114907377-114907377</td>\n",
       "      <td>T</td>\n",
       "      <td>447</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>P/S</td>\n",
       "      <td>chrX:114907377</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.035)</td>\n",
       "      <td>-0.203906</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.56465</td>\n",
       "      <td>9.73350</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.953740</td>\n",
       "      <td>X 114907377 . C T</td>\n",
       "      <td>P</td>\n",
       "      <td>S</td>\n",
       "      <td>5ht2c_human</td>\n",
       "      <td>P28335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15487</th>\n",
       "      <td>654562</td>\n",
       "      <td>X:114907381-114907381</td>\n",
       "      <td>T</td>\n",
       "      <td>448</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>S/F</td>\n",
       "      <td>chrX:114907381</td>\n",
       "      <td>tolerated(0.2)</td>\n",
       "      <td>benign(0.194)</td>\n",
       "      <td>-0.218816</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.00070</td>\n",
       "      <td>19.87100</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.959840</td>\n",
       "      <td>X 114907381 . C T</td>\n",
       "      <td>S</td>\n",
       "      <td>F</td>\n",
       "      <td>5ht2c_human</td>\n",
       "      <td>P28335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15488</th>\n",
       "      <td>654605</td>\n",
       "      <td>X:114907387-114907387</td>\n",
       "      <td>C</td>\n",
       "      <td>450</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>V/A</td>\n",
       "      <td>chrX:114907387</td>\n",
       "      <td>tolerated(0.27)</td>\n",
       "      <td>benign(0.052)</td>\n",
       "      <td>-0.175516</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.63211</td>\n",
       "      <td>6.53650</td>\n",
       "      <td>4.941800e-07</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>0.922960</td>\n",
       "      <td>X 114907387 . T C</td>\n",
       "      <td>V</td>\n",
       "      <td>A</td>\n",
       "      <td>5ht2c_human</td>\n",
       "      <td>P28335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15489</th>\n",
       "      <td>654650</td>\n",
       "      <td>X:114907399-114907399</td>\n",
       "      <td>A</td>\n",
       "      <td>454</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>R/K</td>\n",
       "      <td>chrX:114907399</td>\n",
       "      <td>tolerated(0.67)</td>\n",
       "      <td>benign(0.048)</td>\n",
       "      <td>-0.281812</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.55746</td>\n",
       "      <td>2.88220</td>\n",
       "      <td>3.568200e-06</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0.846640</td>\n",
       "      <td>X 114907399 . G A</td>\n",
       "      <td>R</td>\n",
       "      <td>K</td>\n",
       "      <td>5ht2c_human</td>\n",
       "      <td>P28335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15490</th>\n",
       "      <td>654695</td>\n",
       "      <td>X:114907411-114907411</td>\n",
       "      <td>C</td>\n",
       "      <td>458</td>\n",
       "      <td>HTR2C</td>\n",
       "      <td>V/A</td>\n",
       "      <td>chrX:114907411</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>benign(0.309)</td>\n",
       "      <td>-0.00911951</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.00080</td>\n",
       "      <td>25.76100</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.969010</td>\n",
       "      <td>X 114907411 . T C</td>\n",
       "      <td>V</td>\n",
       "      <td>A</td>\n",
       "      <td>5ht2c_human</td>\n",
       "      <td>P28335</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15491 rows Ã— 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        index                Location Allele  Protein_position SYMBOL  \\\n",
       "0           0  11:113412715-113412715      C               327   DRD2   \n",
       "1          45  11:113412762-113412762      C               311   DRD2   \n",
       "2          90  11:113412766-113412766      A               310   DRD2   \n",
       "3         135  11:113416935-113416935      T               154   DRD2   \n",
       "4         180    13:77901095-77901095      T               305  EDNRB   \n",
       "...       ...                     ...    ...               ...    ...   \n",
       "15486  654517   X:114907377-114907377      T               447  HTR2C   \n",
       "15487  654562   X:114907381-114907381      T               448  HTR2C   \n",
       "15488  654605   X:114907387-114907387      C               450  HTR2C   \n",
       "15489  654650   X:114907399-114907399      A               454  HTR2C   \n",
       "15490  654695   X:114907411-114907411      C               458  HTR2C   \n",
       "\n",
       "      Amino_acids            locus               SIFT  \\\n",
       "0             K/E  chr11:113412715    tolerated(0.27)   \n",
       "1             S/C  chr11:113412762  deleterious(0.05)   \n",
       "2             P/S  chr11:113412766    tolerated(0.61)   \n",
       "3             V/I  chr11:113416935    tolerated(0.16)   \n",
       "4             S/N   chr13:77901095    tolerated(0.92)   \n",
       "...           ...              ...                ...   \n",
       "15486         P/S   chrX:114907377       tolerated(1)   \n",
       "15487         S/F   chrX:114907381     tolerated(0.2)   \n",
       "15488         V/A   chrX:114907387    tolerated(0.27)   \n",
       "15489         R/K   chrX:114907399    tolerated(0.67)   \n",
       "15490         V/A   chrX:114907411     deleterious(0)   \n",
       "\n",
       "                       PolyPhen BayesDel_addAF_score  ...     BETA        SE  \\\n",
       "0                 benign(0.031)            -0.260005  ...  0.43126   0.53696   \n",
       "1      possibly_damaging(0.908)            -0.393485  ... -0.14869   0.18710   \n",
       "2                 benign(0.001)            -0.399698  ... -1.00910   1.63230   \n",
       "3      possibly_damaging(0.767)             0.134914  ... -1.00480   6.30530   \n",
       "4                 benign(0.006)            -0.601516  ... -0.42015   0.22281   \n",
       "...                         ...                  ...  ...      ...       ...   \n",
       "15486             benign(0.035)            -0.203906  ... -0.56465   9.73350   \n",
       "15487             benign(0.194)            -0.218816  ... -1.00070  19.87100   \n",
       "15488             benign(0.052)            -0.175516  ... -0.63211   6.53650   \n",
       "15489             benign(0.048)            -0.281812  ... -0.55746   2.88220   \n",
       "15490             benign(0.309)          -0.00911951  ... -1.00080  25.76100   \n",
       "\n",
       "           AF.Cases AF.Controls    Pvalue                 vcf mutation_from  \\\n",
       "0      3.472200e-03    0.002413  0.421890  11 113412715 . T C             K   \n",
       "1      1.736100e-02    0.020282  0.426780  11 113412762 . G C             S   \n",
       "2      0.000000e+00    0.000298  0.536450  11 113412766 . G A             P   \n",
       "3      0.000000e+00    0.000015  0.873390  11 113416935 . C T             V   \n",
       "4      8.333300e-03    0.014423  0.059333   13 77901095 . C T             S   \n",
       "...             ...         ...       ...                 ...           ...   \n",
       "15486  0.000000e+00    0.000005  0.953740   X 114907377 . C T             P   \n",
       "15487  0.000000e+00    0.000006  0.959840   X 114907381 . C T             S   \n",
       "15488  4.941800e-07    0.000015  0.922960   X 114907387 . T C             V   \n",
       "15489  3.568200e-06    0.000045  0.846640   X 114907399 . G A             R   \n",
       "15490  0.000000e+00    0.000004  0.969010   X 114907411 . T C             V   \n",
       "\n",
       "      mutation_to      protein Uniprot  \n",
       "0               E   drd2_human  P14416  \n",
       "1               C   drd2_human  P14416  \n",
       "2               S   drd2_human  P14416  \n",
       "3               I   drd2_human  P14416  \n",
       "4               N  ednrb_human  P24530  \n",
       "...           ...          ...     ...  \n",
       "15486           S  5ht2c_human  P28335  \n",
       "15487           F  5ht2c_human  P28335  \n",
       "15488           A  5ht2c_human  P28335  \n",
       "15489           K  5ht2c_human  P28335  \n",
       "15490           A  5ht2c_human  P28335  \n",
       "\n",
       "[15491 rows x 68 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "VEPwithGB_icd10 = pd.read_csv('VEPwithGB_icd10_0323.csv', index_col = None)\n",
    "VEPwithGB_icd10['mutation_from'] = VEPwithGB_icd10['Amino_acids'].str.split('/',expand=True)[0]\n",
    "VEPwithGB_icd10['mutation_to'] = VEPwithGB_icd10['Amino_acids'].str.split('/',expand=True)[1]\n",
    "VEPwithGB_icd10_uniid = pd.merge(VEPwithGB_icd10,gene_list_raw[['protein','SYMBOL','Uniprot']],on='SYMBOL',how='left')\n",
    "unq_VEPwithGB_icd10 = VEPwithGB_icd10_uniid.drop_duplicates(subset=['SYMBOL','Protein_position','mutation_from','mutation_to'], keep='first')\n",
    "unq_VEPwithGB_icd10 = unq_VEPwithGB_icd10.reset_index()\n",
    "unq_VEPwithGB_icd10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['P07550', 'P08173', 'P08588', 'P08908', 'P08912', 'P08913',\n",
       "       'P0DMS8', 'P11229', 'P13945', 'P14416', 'P16473', 'P18089',\n",
       "       'P20309', 'P21728', 'P21917', 'P21918', 'P22888', 'P23945',\n",
       "       'P24530', 'P25021', 'P25024', 'P25025', 'P25100', 'P28221',\n",
       "       'P28222', 'P28223', 'P28335', 'P29274', 'P29275', 'P30542',\n",
       "       'P32241', 'P32245', 'P32246', 'P34969', 'P34972', 'P34998',\n",
       "       'P35348', 'P35367', 'P35368', 'P35462', 'P41180', 'P41231',\n",
       "       'P41587', 'P41594', 'P41595', 'P41597', 'P43220', 'P47871',\n",
       "       'P47872', 'P47898', 'P48546', 'P49190', 'P49682', 'P50406',\n",
       "       'P51677', 'P51681', 'P51685', 'P61073', 'Q02643', 'Q03431',\n",
       "       'Q13255', 'Q13639', 'Q14416', 'Q16602', 'Q7RTX0', 'Q9H244',\n",
       "       'Q9H3N8', 'Q9Y5N1'], dtype=object)"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniid_list = np.unique(unq_VEPwithGB_icd10['Uniprot'])\n",
    "uniid_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# for each GPCR, get AF pdb file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# a function to get the alphafold structure (pdb file) for each gene\n",
    "def getAF(uniprot_id):\n",
    "    # Set the API endpoint and the UniProt ID of the protein\n",
    "    endpoint = \"https://alphafold.ebi.ac.uk/files\"\n",
    "\n",
    "    # Send a GET request to the API to retrieve the protein's structure\n",
    "    response = requests.get(f\"{endpoint}/AF-{uniprot_id}-F1-model_v4.pdb\")\n",
    "\n",
    "    # Check the status code of the response to ensure it was successful\n",
    "    if response.status_code == 200:\n",
    "        filename = f\"/Users/joannahench/Downloads/foldx5MacStd.tar_/Genebass/{uniprot_id}/{uniprot_id}.pdb\"\n",
    "        os.makedirs(os.path.dirname(filename), exist_ok=True)\n",
    "        # Write the contents of the response to a PDB file\n",
    "        with open(filename, \"w\") as f:\n",
    "            f.write(response.text)\n",
    "    else:\n",
    "        print(f\"Failed to retrieve structure for {uniprot_id}: {response.status_code}\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "#get the structure\n",
    "for i in range(len(uniid_list)):\n",
    "    getAF(uniid_list[i])"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# make mutation list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "for i in range(unq_VEPwithGB_icd10.shape[0]):\n",
    "    # get the uniprot id according to entryname and convert the Series to a string with index excluded\n",
    "    uniprot_id = unq_VEPwithGB_icd10['Uniprot'][i]\n",
    "    # gene = gene_list_raw.loc[gene_list_raw['protein']==gpcr_valid_missense.iloc[i,4],'gene'].to_string(index=False)\n",
    "    # uniid_list.append(uniprot_id)\n",
    "    # gene_list.append(gene)\n",
    "    mut_info = unq_VEPwithGB_icd10.iloc[i,64]+'A'+str(unq_VEPwithGB_icd10.iloc[i,3])+unq_VEPwithGB_icd10.iloc[i,65]\n",
    "    with open(f\"/Users/joannahench/Downloads/foldx5MacStd.tar_/Genebass/{uniprot_id}/individual_list_{uniprot_id}.txt\", 'a') as file:\n",
    "        # write a new line to the file\n",
    "        file.write(mut_info+';'+\"\\n\")\n",
    "    "
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# make bash file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# generate the bash file\n",
    "for i in range(len(uniid_list)):\n",
    "        uniprot_id = uniid_list[i]\n",
    "        with open(\"foldx_command.sh\", 'a') as foldx:\n",
    "                foldx.write(f'cd {uniprot_id}'+\"\\n\")\n",
    "                # write the command\n",
    "                foldx.write(f\"../../foldx --command=BuildModel --pdb={uniprot_id}.pdb --mutant-file=individual_list_{uniprot_id}.txt\"+\"\\n\")\n",
    "                foldx.write('cd ..'+\"\\n\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "../../foldx --help --command=BuildModel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "../../foldx --command=BuildModel --pdb=/Users/joannahench/Downloads/foldx5MacStd.tar_/GPCRdb/P0DMS8/P0DMS8.pdb --mutant-file=/Users/joannahench/Downloads/foldx5MacStd.tar_/GPCRdb/P0DMS8/individual_list_P0DMS8.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "nohup ./foldx_test.sh &"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
