{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# README  \n",
    "This notebook mainly did two things:  \n",
    "Given the 400 olfactory-related GPCRs, extract their mutation data from GPCRdb. - get in vitro data  \n",
    "At the same time, run Ensembl VEPs on the mutation data. - get VEP scores"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# preliminary "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import pandas as pd\n",
    "import os\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import re\n",
    "import requests\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 20 types of amino acids\n",
    "AA_all = {\n",
    "    \"A\" : \"Ala\",\n",
    "    \"R\" : \"Arg\",\n",
    "    \"N\" : \"Asn\",\n",
    "    \"D\" : \"Asp\",\n",
    "    \"C\" : \"Cys\",\n",
    "    \"E\" : \"Glu\",\n",
    "    \"Q\" : \"Gln\",\n",
    "    \"G\" : \"Gly\",\n",
    "    \"H\" : \"His\",\n",
    "    \"I\" : \"Ile\",\n",
    "    \"L\" : \"Leu\",\n",
    "    \"K\" : \"Lys\",\n",
    "    \"M\" : \"Met\",\n",
    "    \"F\" : \"Phe\",\n",
    "    \"P\" : \"Pro\",\n",
    "    \"S\" : \"Ser\",\n",
    "    \"T\" : \"Thr\",\n",
    "    \"W\" : \"Trp\",\n",
    "    \"Y\" : \"Tyr\",\n",
    "    \"V\" : \"Val\"\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read in all gene names\n",
    "gene_list_raw =pd.read_csv(\"230323_EST_ENSG_GENE_new.csv\")\n",
    "# gene_list_raw\n",
    "protein_list = gene_list_raw[\"protein\"].tolist()\n",
    "enst_list = gene_list_raw[\"ENST\"].tolist()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>ENST</th>\n",
       "      <th>ENSG</th>\n",
       "      <th>protein</th>\n",
       "      <th>gene</th>\n",
       "      <th>Uniprot</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>ENST00000646641</td>\n",
       "      <td>ENSG00000267534</td>\n",
       "      <td>s1pr2_human</td>\n",
       "      <td>S1PR2</td>\n",
       "      <td>O95136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>121</td>\n",
       "      <td>ENST00000547270</td>\n",
       "      <td>ENSG00000257138</td>\n",
       "      <td>t2r38_human</td>\n",
       "      <td>TAS2R38</td>\n",
       "      <td>P59533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>219</td>\n",
       "      <td>ENST00000390675</td>\n",
       "      <td>ENSG00000256436</td>\n",
       "      <td>t2r31_human</td>\n",
       "      <td>TAS2R31</td>\n",
       "      <td>P59538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>344</td>\n",
       "      <td>ENST00000539585</td>\n",
       "      <td>ENSG00000256188</td>\n",
       "      <td>t2r30_human</td>\n",
       "      <td>TAS2R30</td>\n",
       "      <td>P59541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>483</td>\n",
       "      <td>ENST00000538986</td>\n",
       "      <td>ENSG00000255837</td>\n",
       "      <td>t2r20_human</td>\n",
       "      <td>TAS2R20</td>\n",
       "      <td>P59543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>64029</td>\n",
       "      <td>ENST00000510937</td>\n",
       "      <td>ENSG00000226306</td>\n",
       "      <td>npy6r_human</td>\n",
       "      <td>NPY6R</td>\n",
       "      <td>Q99463</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>64030</td>\n",
       "      <td>ENST00000641193</td>\n",
       "      <td>ENSG00000279301</td>\n",
       "      <td>o2t11_human</td>\n",
       "      <td>OR2T11</td>\n",
       "      <td>Q8NH01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>397</th>\n",
       "      <td>64031</td>\n",
       "      <td>ENST00000641732</td>\n",
       "      <td>ENSG00000172146</td>\n",
       "      <td>or1a1_human</td>\n",
       "      <td>OR1A1</td>\n",
       "      <td>Q9P1Q5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>398</th>\n",
       "      <td>64032</td>\n",
       "      <td>ENST00000328890</td>\n",
       "      <td>ENSG00000183024</td>\n",
       "      <td>or1g1_human</td>\n",
       "      <td>OR1G1</td>\n",
       "      <td>P47890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399</th>\n",
       "      <td>64034</td>\n",
       "      <td>ENST00000466706</td>\n",
       "      <td>ENSG00000179073</td>\n",
       "      <td>taar3_human</td>\n",
       "      <td>TAAR3P</td>\n",
       "      <td>Q9P1P4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>400 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     index             ENST             ENSG      protein     gene Uniprot\n",
       "0        0  ENST00000646641  ENSG00000267534  s1pr2_human    S1PR2  O95136\n",
       "1      121  ENST00000547270  ENSG00000257138  t2r38_human  TAS2R38  P59533\n",
       "2      219  ENST00000390675  ENSG00000256436  t2r31_human  TAS2R31  P59538\n",
       "3      344  ENST00000539585  ENSG00000256188  t2r30_human  TAS2R30  P59541\n",
       "4      483  ENST00000538986  ENSG00000255837  t2r20_human  TAS2R20  P59543\n",
       "..     ...              ...              ...          ...      ...     ...\n",
       "395  64029  ENST00000510937  ENSG00000226306  npy6r_human    NPY6R  Q99463\n",
       "396  64030  ENST00000641193  ENSG00000279301  o2t11_human   OR2T11  Q8NH01\n",
       "397  64031  ENST00000641732  ENSG00000172146  or1a1_human    OR1A1  Q9P1Q5\n",
       "398  64032  ENST00000328890  ENSG00000183024  or1g1_human    OR1G1  P47890\n",
       "399  64034  ENST00000466706  ENSG00000179073  taar3_human   TAAR3P  Q9P1P4\n",
       "\n",
       "[400 rows x 6 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_list_raw"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# gname = gene_list[124]\n",
    "# # gname = 'o2t11_human'\n",
    "# # enst = enst_list [0]\n",
    "# url = 'https://gpcrdb.org/services/mutants/'+gname\n",
    "# response = requests.get(url)\n",
    "# mutants_data = response.json()\n",
    "# df = pd.DataFrame(mutants_data)\n",
    "# df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "s1pr2_human info not found!\n",
      "t2r38_human info not found!\n",
      "t2r31_human info not found!\n",
      "t2r30_human info not found!\n",
      "t2r20_human info not found!\n",
      "hcar3_human info not found!\n",
      "t2r43_human info not found!\n",
      "taar9_human info not found!\n",
      "t2r39_human info not found!\n",
      "t2r46_human info not found!\n",
      "t2r40_human info not found!\n",
      "t2r41_human info not found!\n",
      "gpr33_human info not found!\n",
      "s1pr3_human info not found!\n",
      "t2r13_human info not found!\n",
      "t2r14_human info not found!\n",
      "t2r19_human info not found!\n",
      "lgr4_human info not found!\n",
      "gpr20_human info not found!\n",
      "npy4r_human info not found!\n",
      "gpr52_human info not found!\n",
      "sucr1_human info not found!\n",
      "v1br_human info not found!\n",
      "hcar1_human info not found!\n",
      "gpr21_human info not found!\n",
      "fpr3_human info not found!\n",
      "gp141_human info not found!\n",
      "ffar4_human info not found!\n",
      "t2r42_human info not found!\n",
      "ffar3_human info not found!\n",
      "lpar5_human info not found!\n",
      "ccr10_human info not found!\n",
      "gp173_human info not found!\n",
      "gpr39_human info not found!\n",
      "ccr4_human info not found!\n",
      "npbw1_human info not found!\n",
      "mrgx2_human info not found!\n",
      "gpr1_human info not found!\n",
      "gpr19_human info not found!\n",
      "pd2r2_human info not found!\n",
      "hcar2_human info not found!\n",
      "galr2_human info not found!\n",
      "mrgrg_human info not found!\n",
      "p2ry8_human info not found!\n",
      "gpr3_human info not found!\n",
      "gpr88_human info not found!\n",
      "p2y13_human info not found!\n",
      "gp135_human info not found!\n",
      "ur2r_human info not found!\n",
      "gpr62_human info not found!\n",
      "gp157_human info not found!\n",
      "ssr2_human info not found!\n",
      "fzd2_human info not found!\n",
      "gp139_human info not found!\n",
      "agrd2_human info not found!\n",
      "gpbar_human info not found!\n",
      "mrgx3_human info not found!\n",
      "mrgx4_human info not found!\n",
      "ts1r2_human info not found!\n",
      "gp150_human info not found!\n",
      "gpr4_human info not found!\n",
      "fzd8_human info not found!\n",
      "gp156_human info not found!\n",
      "gp152_human info not found!\n",
      "gp149_human info not found!\n",
      "gp171_human info not found!\n",
      "p2y14_human info not found!\n",
      "fzd4_human info not found!\n",
      "cml1_human info not found!\n",
      "gp160_human info not found!\n",
      "gpc6a_human info not found!\n",
      "xcr1_human info not found!\n",
      "gp148_human info not found!\n",
      "gp151_human info not found!\n",
      "rl3r2_human info not found!\n",
      "mrgrd_human info not found!\n",
      "cxcr6_human info not found!\n",
      "gpr22_human info not found!\n",
      "gpr34_human info not found!\n",
      "gpr82_human info not found!\n",
      "nmur1_human info not found!\n",
      "lpar3_human info not found!\n",
      "s1pr1_human info not found!\n",
      "gpr27_human info not found!\n",
      "mrgx1_human info not found!\n",
      "gpr25_human info not found!\n",
      "nk3r_human info not found!\n",
      "ta2r1_human info not found!\n",
      "pkr1_human info not found!\n",
      "gp183_human info not found!\n",
      "ntr2_human info not found!\n",
      "pd2r_human info not found!\n",
      "oxgr1_human info not found!\n",
      "gp101_human info not found!\n",
      "gper1_human info not found!\n",
      "gp146_human info not found!\n",
      "gpr85_human info not found!\n",
      "npy5r_human info not found!\n",
      "fzd5_human info not found!\n",
      "oxer1_human info not found!\n",
      "agrl4_human info not found!\n",
      "pe2r1_human info not found!\n",
      "cxcr5_human info not found!\n",
      "agrg5_human info not found!\n",
      "gp153_human info not found!\n",
      "fzd1_human info not found!\n",
      "agrg4_human info not found!\n",
      "gpr61_human info not found!\n",
      "fzd7_human info not found!\n",
      "gpr78_human info not found!\n",
      "gpr26_human info not found!\n",
      "gpr15_human info not found!\n",
      "cltr2_human info not found!\n",
      "mchr2_human info not found!\n",
      "gp158_human info not found!\n",
      "gp107_human info not found!\n",
      "gp119_human info not found!\n",
      "taar8_human info not found!\n",
      "taar6_human info not found!\n",
      "taar2_human info not found!\n",
      "ackr2_human info not found!\n",
      "ackr3_human info not found!\n",
      "gpr32_human info not found!\n",
      "psyr_human info not found!\n",
      "lpar6_human info not found!\n",
      "gpr84_human info not found!\n",
      "lgr5_human info not found!\n",
      "gpr87_human info not found!\n",
      "ox2r_human info not found!\n",
      "gpr45_human info not found!\n",
      "gpr55_human info not found!\n",
      "nmbr_human info not found!\n",
      "taar5_human info not found!\n",
      "c5ar2_human info not found!\n",
      "apj_human info not found!\n",
      "rxfp2_human info not found!\n",
      "gpr12_human info not found!\n",
      "nmur2_human info not found!\n",
      "ackr4_human info not found!\n",
      "galr3_human info not found!\n",
      "ta2r5_human info not found!\n",
      "ta2r4_human info not found!\n",
      "ta2r3_human info not found!\n",
      "ccr7_human info not found!\n",
      "ffar2_human info not found!\n",
      "grpr_human info not found!\n",
      "s1pr4_human info not found!\n",
      "npbw2_human info not found!\n",
      "gpr18_human info not found!\n",
      "opn5_human info not found!\n",
      "gpr83_human info not found!\n",
      "ta2r9_human info not found!\n",
      "ta2r7_human info not found!\n",
      "t2r10_human info not found!\n",
      "ta2r8_human info not found!\n",
      "gpr31_human info not found!\n",
      "prlhr_human info not found!\n",
      "gpr75_human info not found!\n",
      "grm6_human info not found!\n",
      "gpr63_human info not found!\n",
      "fzd3_human info not found!\n",
      "mtr1l_human info not found!\n",
      "pkr2_human info not found!\n",
      "p2y10_human info not found!\n",
      "glp2r_human info not found!\n",
      "lpar2_human info not found!\n",
      "agrg1_human info not found!\n",
      "gp176_human info not found!\n",
      "tbb3_human info not found!\n",
      "ogr1_human info not found!\n",
      "lt4r2_human info not found!\n",
      "fzd6_human info not found!\n",
      "agrb1_human info not found!\n",
      "tpra1_human info not found!\n",
      "agra2_human info not found!\n",
      "agrv1_human info not found!\n",
      "qrfpr_human info not found!\n",
      "agra1_human info not found!\n",
      "mrgre_human info not found!\n",
      "lpar1_human info not found!\n",
      "ox1r_human info not found!\n",
      "agrb2_human info not found!\n",
      "agrf1_human info not found!\n",
      "pe2r3_human info not found!\n",
      "agrl2_human info not found!\n",
      "agrb3_human info not found!\n",
      "gp161_human info not found!\n",
      "lgr6_human info not found!\n",
      "opn3_human info not found!\n",
      "grm3_human info not found!\n",
      "npsr1_human info not found!\n",
      "grm7_human info not found!\n",
      "ccr9_human info not found!\n",
      "grm8_human info not found!\n",
      "gp142_human info not found!\n",
      "agra3_human info not found!\n",
      "s1pr5_human info not found!\n",
      "gp132_human info not found!\n",
      "agrf4_human info not found!\n",
      "gpr35_human info not found!\n",
      "agre1_human info not found!\n",
      "agrf3_human info not found!\n",
      "gp162_human info not found!\n",
      "pacr_human info not found!\n",
      "gpc5b_human info not found!\n",
      "agrf2_human info not found!\n",
      "gpr6_human info not found!\n",
      "agrf5_human info not found!\n",
      "celr1_human info not found!\n",
      "agrd1_human info not found!\n",
      "agre3_human info not found!\n",
      "par4_human info not found!\n",
      "kissr_human info not found!\n",
      "fzd10_human info not found!\n",
      "gpc5d_human info not found!\n",
      "mtlr_human info not found!\n",
      "rai3_human info not found!\n",
      "agrg6_human info not found!\n",
      "agrl1_human info not found!\n",
      "ts1r1_human info not found!\n",
      "npff2_human info not found!\n",
      "g137a_human info not found!\n",
      "agrl3_human info not found!\n",
      "gpc5c_human info not found!\n",
      "t2r50_human info not found!\n",
      "agrg3_human info not found!\n",
      "agre5_human info not found!\n",
      "gp179_human info not found!\n",
      "agre4_human info not found!\n",
      "npy6r_human info not found!\n",
      "o2t11_human info not found!\n",
      "or1a1_human info not found!\n",
      "or1g1_human info not found!\n",
      "taar3_human info not found!\n",
      "there are 166 genes have mutants info\n"
     ]
    }
   ],
   "source": [
    "# check how many proteins have mutants info \n",
    "count = 0\n",
    "mutant_genes =[]\n",
    "for i in range(len(protein_list)):\n",
    "    gname = protein_list[i]\n",
    "    url = 'https://gpcrdb.org/services/mutants/'+gname\n",
    "    response = requests.get(url)\n",
    "    mutants_data = response.json()\n",
    "    if(len(mutants_data)>0):\n",
    "      count = count + 1\n",
    "      mutant_genes.append(gname)\n",
    "    else:\n",
    "      print(gname+' info not found!')\n",
    "\n",
    "print('there are '+str(len(mutant_genes))+' genes have mutants info')"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "there are 166 genes have mutants info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>ENST</th>\n",
       "      <th>ENSG</th>\n",
       "      <th>protein</th>\n",
       "      <th>gene</th>\n",
       "      <th>Uniprot</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>838</td>\n",
       "      <td>ENST00000321826</td>\n",
       "      <td>ENSG00000244165</td>\n",
       "      <td>p2y11_human</td>\n",
       "      <td>P2RY11</td>\n",
       "      <td>Q96G91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1668</td>\n",
       "      <td>ENST00000345363</td>\n",
       "      <td>ENSG00000213903</td>\n",
       "      <td>lt4r1_human</td>\n",
       "      <td>LTB4R</td>\n",
       "      <td>Q15722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2679</td>\n",
       "      <td>ENST00000377034</td>\n",
       "      <td>ENSG00000204681</td>\n",
       "      <td>gabr1_human</td>\n",
       "      <td>GABBR1</td>\n",
       "      <td>Q9UBS5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3455</td>\n",
       "      <td>ENST00000355085</td>\n",
       "      <td>ENSG00000197405</td>\n",
       "      <td>c5ar1_human</td>\n",
       "      <td>C5AR1</td>\n",
       "      <td>P21730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3758</td>\n",
       "      <td>ENST00000438284</td>\n",
       "      <td>ENSG00000196639</td>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>HRH1</td>\n",
       "      <td>P35367</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>62409</td>\n",
       "      <td>ENST00000337539</td>\n",
       "      <td>ENSG00000128271</td>\n",
       "      <td>aa2ar_human</td>\n",
       "      <td>ADORA2A</td>\n",
       "      <td>P29274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>62533</td>\n",
       "      <td>ENST00000620793</td>\n",
       "      <td>ENSG00000274286</td>\n",
       "      <td>ada2b_human</td>\n",
       "      <td>ADRA2B</td>\n",
       "      <td>P18089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>62696</td>\n",
       "      <td>ENST00000610913</td>\n",
       "      <td>ENSG00000278195</td>\n",
       "      <td>ssr3_human</td>\n",
       "      <td>SSTR3</td>\n",
       "      <td>P32745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>62881</td>\n",
       "      <td>ENST00000280155</td>\n",
       "      <td>ENSG00000150594</td>\n",
       "      <td>ada2a_human</td>\n",
       "      <td>ADRA2A</td>\n",
       "      <td>P08913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>64027</td>\n",
       "      <td>ENST00000241356</td>\n",
       "      <td>ENSG00000282608</td>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>ADORA3</td>\n",
       "      <td>P0DMS8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>166 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     index             ENST             ENSG      protein     gene Uniprot\n",
       "0      838  ENST00000321826  ENSG00000244165  p2y11_human   P2RY11  Q96G91\n",
       "1     1668  ENST00000345363  ENSG00000213903  lt4r1_human    LTB4R  Q15722\n",
       "2     2679  ENST00000377034  ENSG00000204681  gabr1_human   GABBR1  Q9UBS5\n",
       "3     3455  ENST00000355085  ENSG00000197405  c5ar1_human    C5AR1  P21730\n",
       "4     3758  ENST00000438284  ENSG00000196639   hrh1_human     HRH1  P35367\n",
       "..     ...              ...              ...          ...      ...     ...\n",
       "161  62409  ENST00000337539  ENSG00000128271  aa2ar_human  ADORA2A  P29274\n",
       "162  62533  ENST00000620793  ENSG00000274286  ada2b_human   ADRA2B  P18089\n",
       "163  62696  ENST00000610913  ENSG00000278195   ssr3_human    SSTR3  P32745\n",
       "164  62881  ENST00000280155  ENSG00000150594  ada2a_human   ADRA2A  P08913\n",
       "165  64027  ENST00000241356  ENSG00000282608   aa3r_human   ADORA3  P0DMS8\n",
       "\n",
       "[166 rows x 6 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# only extract the info of those 167 genes\n",
    "mutant_genes_df = gene_list_raw[gene_list_raw['protein'].isin(mutant_genes)]\n",
    "mutant_genes_df.reset_index(drop=True, inplace=True)\n",
    "mutant_genes_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# in 166 genes, find if their enst have info in ensemble(because later we need to use enst id as input to ensembl vep)\n",
    "server = \"https://rest.ensembl.org\"\n",
    "for i in range(mutant_genes_df.shape[0]):\n",
    "    gname = mutant_genes_df['gene'][i]\n",
    "    enst = mutant_genes_df['ENST'][i]\n",
    "    # ensg = mutant_genes_df['ENSG'][i]\n",
    "    ext = \"/sequence/id/\"+enst+\"?type=cdna\"\n",
    "    r = requests.get(server+ext, headers={ \"Content-Type\" : \"text/x-fasta\"})\n",
    "    if not r.ok:\n",
    "      print(gname+' ENST not OK')"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get access to GPCRdb"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "all ENSTs are OK! Now next extract mutation data. create two dataframes: gpcr_all_mutants stores all data from GPCRsb, vep_input all stores the mutation info(but transform from 1 letter abbreviation to 3 letter abbreviation) as the further inputs to Ensembl VEP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "processing No.0 ---> p2y11_human\n",
      "processing No.1 ---> lt4r1_human\n",
      "processing No.2 ---> gabr1_human\n",
      "processing No.3 ---> c5ar1_human\n",
      "processing No.4 ---> hrh1_human\n",
      "processing No.5 ---> cnr2_human\n",
      "processing No.6 ---> adrb3_human\n",
      "processing No.7 ---> fzd9_human\n",
      "processing No.8 ---> p2ry4_human\n",
      "processing No.9 ---> t2r60_human\n",
      "processing No.10 ---> acthr_human\n",
      "processing No.11 ---> npy2r_human\n",
      "processing No.12 ---> acm5_human\n",
      "processing No.13 ---> drd1_human\n",
      "processing No.14 ---> rl3r1_human\n",
      "processing No.15 ---> acm2_human\n",
      "processing No.16 ---> oxyr_human\n",
      "processing No.17 ---> cxcr2_human\n",
      "processing No.18 ---> agtr2_human\n",
      "processing No.19 ---> acm4_human\n",
      "processing No.20 ---> 5ht1d_human\n",
      "processing No.21 ---> 5ht1f_human\n",
      "processing No.22 ---> 5ht1a_human\n",
      "processing No.23 ---> mc5r_human\n",
      "processing No.24 ---> p2ry2_human\n",
      "processing No.25 ---> trfr_human\n",
      "processing No.26 ---> agrg2_human\n",
      "processing No.27 ---> cltr1_human\n",
      "processing No.28 ---> ada1d_human\n",
      "processing No.29 ---> c3ar_human\n",
      "processing No.30 ---> p2ry6_human\n",
      "processing No.31 ---> pe2r4_human\n",
      "processing No.32 ---> fpr1_human\n",
      "processing No.33 ---> fpr2_human\n",
      "processing No.34 ---> gpr37_human\n",
      "processing No.35 ---> aa2br_human\n",
      "processing No.36 ---> ada1b_human\n",
      "processing No.37 ---> g37l1_human\n",
      "processing No.38 ---> ts1r3_human\n",
      "processing No.39 ---> p2ry1_human\n",
      "processing No.40 ---> drd5_human\n",
      "processing No.41 ---> ptafr_human\n",
      "processing No.42 ---> p2y12_human\n",
      "processing No.43 ---> adrb2_human\n",
      "processing No.44 ---> 5ht1e_human\n",
      "processing No.45 ---> acm1_human\n",
      "processing No.46 ---> mtr1a_human\n",
      "processing No.47 ---> bkrb2_human\n",
      "processing No.48 ---> gp182_human\n",
      "processing No.49 ---> mc4r_human\n",
      "processing No.50 ---> galr1_human\n",
      "processing No.51 ---> par3_human\n",
      "processing No.52 ---> npy1r_human\n",
      "processing No.53 ---> grm2_human\n",
      "processing No.54 ---> opsd_human\n",
      "processing No.55 ---> ccr1_human\n",
      "processing No.56 ---> aa1r_human\n",
      "processing No.57 ---> cxcr1_human\n",
      "processing No.58 ---> cckar_human\n",
      "processing No.59 ---> ccr5_human\n",
      "processing No.60 ---> 5ht6r_human\n",
      "processing No.61 ---> 5ht5a_human\n",
      "processing No.62 ---> drd2_human\n",
      "processing No.63 ---> npff1_human\n",
      "processing No.64 ---> lpar4_human\n",
      "processing No.65 ---> gp174_human\n",
      "processing No.66 ---> taar1_human\n",
      "processing No.67 ---> agtr1_human\n",
      "processing No.68 ---> agrg7_human\n",
      "processing No.69 ---> pth2r_human\n",
      "processing No.70 ---> gpr17_human\n",
      "processing No.71 ---> celr2_human\n",
      "processing No.72 ---> ssr1_human\n",
      "processing No.73 ---> lshr_human\n",
      "processing No.74 ---> gabr2_human\n",
      "processing No.75 ---> 5ht2b_human\n",
      "processing No.76 ---> 5ht1b_human\n",
      "processing No.77 ---> hrh4_human\n",
      "processing No.78 ---> acm3_human\n",
      "processing No.79 ---> ssr4_human\n",
      "processing No.80 ---> mas_human\n",
      "processing No.81 ---> opsb_human\n",
      "processing No.82 ---> smo_human\n",
      "processing No.83 ---> t2r16_human\n",
      "processing No.84 ---> mchr1_human\n",
      "processing No.85 ---> ffar1_human\n",
      "processing No.86 ---> gpr42_human\n",
      "processing No.87 ---> oprx_human\n",
      "processing No.88 ---> pe2r2_human\n",
      "processing No.89 ---> mc3r_human\n",
      "processing No.90 ---> oprd_human\n",
      "processing No.91 ---> nk1r_human\n",
      "processing No.92 ---> ccr6_human\n",
      "processing No.93 ---> glp1r_human\n",
      "processing No.94 ---> brs3_human\n",
      "processing No.95 ---> opsr_human\n",
      "processing No.96 ---> ntr1_human\n",
      "processing No.97 ---> bkrb1_human\n",
      "processing No.98 ---> sctr_human\n",
      "processing No.99 ---> nk2r_human\n",
      "processing No.100 ---> calrl_human\n",
      "processing No.101 ---> celr3_human\n",
      "processing No.102 ---> gipr_human\n",
      "processing No.103 ---> mshr_human\n",
      "processing No.104 ---> grm4_human\n",
      "processing No.105 ---> crfr2_human\n",
      "processing No.106 ---> gp143_human\n",
      "processing No.107 ---> pth1r_human\n",
      "processing No.108 ---> ccr2_human\n",
      "processing No.109 ---> glr_human\n",
      "processing No.110 ---> cx3c1_human\n",
      "processing No.111 ---> ccrl2_human\n",
      "processing No.112 ---> drd3_human\n",
      "processing No.113 ---> 5ht2a_human\n",
      "processing No.114 ---> ta2r_human\n",
      "processing No.115 ---> cxcr3_human\n",
      "processing No.116 ---> pf2r_human\n",
      "processing No.117 ---> cnr1_human\n",
      "processing No.118 ---> adrb1_human\n",
      "processing No.119 ---> ackr1_human\n",
      "processing No.120 ---> v2r_human\n",
      "processing No.121 ---> ccr3_human\n",
      "processing No.122 ---> hrh3_human\n",
      "processing No.123 ---> 5ht7r_human\n",
      "processing No.124 ---> gasr_human\n",
      "processing No.125 ---> ada2c_human\n",
      "processing No.126 ---> ccr8_human\n",
      "processing No.127 ---> ghrhr_human\n",
      "processing No.128 ---> vipr1_human\n",
      "processing No.129 ---> ednra_human\n",
      "processing No.130 ---> par1_human\n",
      "processing No.131 ---> agre2_human\n",
      "processing No.132 ---> mrgrf_human\n",
      "processing No.133 ---> rxfp1_human\n",
      "processing No.134 ---> grm5_human\n",
      "processing No.135 ---> v1ar_human\n",
      "processing No.136 ---> par2_human\n",
      "processing No.137 ---> casr_human\n",
      "processing No.138 ---> ssr5_human\n",
      "processing No.139 ---> pi2r_human\n",
      "processing No.140 ---> grm1_human\n",
      "processing No.141 ---> ada1a_human\n",
      "processing No.142 ---> 5ht2c_human\n",
      "processing No.143 ---> oprk_human\n",
      "processing No.144 ---> vipr2_human\n",
      "processing No.145 ---> mtr1b_human\n",
      "processing No.146 ---> opn4_human\n",
      "processing No.147 ---> cxcr4_human\n",
      "processing No.148 ---> ghsr_human\n",
      "processing No.149 ---> hrh2_human\n",
      "processing No.150 ---> gnrhr_human\n",
      "processing No.151 ---> crfr1_human\n",
      "processing No.152 ---> oprm_human\n",
      "processing No.153 ---> tshr_human\n",
      "processing No.154 ---> fshr_human\n",
      "processing No.155 ---> calcr_human\n",
      "processing No.156 ---> drd4_human\n",
      "processing No.157 ---> 5ht4r_human\n",
      "processing No.158 ---> ednrb_human\n",
      "processing No.159 ---> mas1l_human\n",
      "processing No.160 ---> opsg_human\n",
      "processing No.161 ---> aa2ar_human\n",
      "processing No.162 ---> ada2b_human\n",
      "processing No.163 ---> ssr3_human\n",
      "processing No.164 ---> ada2a_human\n",
      "processing No.165 ---> aa3r_human\n"
     ]
    }
   ],
   "source": [
    "gpcr_all_mutants = pd.DataFrame()\n",
    "vep_input_all = pd.DataFrame()\n",
    "for i in range(mutant_genes_df.shape[0]):\n",
    "  gname = mutant_genes_df['protein'][i]\n",
    "  enst = mutant_genes_df['ENST'][i]\n",
    "  url = 'https://gpcrdb.org/services/mutants/'+gname\n",
    "  response = requests.get(url)\n",
    "  mutants_data = response.json()\n",
    "  df = pd.DataFrame(mutants_data)\n",
    "  # only keep missense mutation\n",
    "  df = df[df['mutation_to']!='-']\n",
    "  df = df[df['mutation_to']!='X']\n",
    "  df = df[df['mutation_to']!='']\n",
    "  df = df[df['mutation_from']!=df['mutation_to']]\n",
    "  df.reset_index(drop=True, inplace=True)\n",
    "  gpcr_all_mutants = gpcr_all_mutants.append(df)\n",
    "  vep_input_df = pd.DataFrame()\n",
    "  print(\"processing No.\"+str(i)+\" ---> \"+gname)\n",
    "  for j in range(df.shape[0]):\n",
    "    row = vep_input_df.shape[0]\n",
    "    vep_input_df.loc[row,'protein'] = gname\n",
    "    vep_input_df.loc[row,'ENST'] = enst\n",
    "    vep_input_df.loc[row,'mutation_pos'] = df.loc[j,\"mutation_pos\"]\n",
    "    # we need to store this column so that we can filter out rows with no exp_fold_change value\n",
    "    vep_input_df.loc[row,'exp_fold_change'] = df.loc[j,\"exp_fold_change\"]\n",
    "    vep_input_df.loc[row,'mutation_from'] = AA_all[df.loc[j,\"mutation_from\"]] \n",
    "    vep_input_df.loc[row,'mutation_to'] = AA_all[df.loc[j,\"mutation_to\"]]\n",
    "  vep_input_df['mutation_pos'] = vep_input_df['mutation_pos'].astype(int) \n",
    "  vep_input_df['vep_input'] = enst + ':p.' + vep_input_df['mutation_from'] + vep_input_df['mutation_pos'].astype(str) + vep_input_df['mutation_to']\n",
    "  \n",
    "  vep_input_all = vep_input_all.append(vep_input_df)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(24455, 7)\n",
      "(24455, 25)\n"
     ]
    }
   ],
   "source": [
    "print(vep_input_all.shape)\n",
    "print(gpcr_all_mutants.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>reference</th>\n",
       "      <th>protein</th>\n",
       "      <th>mutation_pos</th>\n",
       "      <th>mutation_from</th>\n",
       "      <th>mutation_to</th>\n",
       "      <th>ligand_name</th>\n",
       "      <th>ligand_idtype</th>\n",
       "      <th>ligand_id</th>\n",
       "      <th>ligand_class</th>\n",
       "      <th>exp_type</th>\n",
       "      <th>...</th>\n",
       "      <th>exp_mu_effect_value</th>\n",
       "      <th>exp_fold_change</th>\n",
       "      <th>exp_mu_effect_qual</th>\n",
       "      <th>exp_mu_effect_ligand_prop</th>\n",
       "      <th>exp_mu_ligand_ref</th>\n",
       "      <th>opt_receptor_expression</th>\n",
       "      <th>opt_basal_activity</th>\n",
       "      <th>opt_gain_of_activity</th>\n",
       "      <th>opt_ligand_emax</th>\n",
       "      <th>opt_agonist</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>15626750</td>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>37</td>\n",
       "      <td>I</td>\n",
       "      <td>V</td>\n",
       "      <td>Cetirizine</td>\n",
       "      <td>ChEMBL Compound ID</td>\n",
       "      <td>CHEMBL1000</td>\n",
       "      <td>Binding - unknown pharmacological activity</td>\n",
       "      <td>pK(i)</td>\n",
       "      <td>...</td>\n",
       "      <td>7.700</td>\n",
       "      <td>-1.585</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[3H]-mepyramine (radioligand)</td>\n",
       "      <td>94.2</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>15626750</td>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>38</td>\n",
       "      <td>C</td>\n",
       "      <td>S</td>\n",
       "      <td>Cetirizine</td>\n",
       "      <td>ChEMBL Compound ID</td>\n",
       "      <td>CHEMBL1000</td>\n",
       "      <td>Binding - unknown pharmacological activity</td>\n",
       "      <td>pK(i)</td>\n",
       "      <td>...</td>\n",
       "      <td>7.900</td>\n",
       "      <td>1.000</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[3H]-mepyramine (radioligand)</td>\n",
       "      <td>141.3</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>15626750</td>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>84</td>\n",
       "      <td>N</td>\n",
       "      <td>S</td>\n",
       "      <td>Cetirizine</td>\n",
       "      <td>ChEMBL Compound ID</td>\n",
       "      <td>CHEMBL1000</td>\n",
       "      <td>Binding - unknown pharmacological activity</td>\n",
       "      <td>pK(i)</td>\n",
       "      <td>...</td>\n",
       "      <td>7.500</td>\n",
       "      <td>-2.513</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[3H]-mepyramine (radioligand)</td>\n",
       "      <td>53.7</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>16408006</td>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>111</td>\n",
       "      <td>S</td>\n",
       "      <td>A</td>\n",
       "      <td>Cetirizine</td>\n",
       "      <td>ChEMBL Compound ID</td>\n",
       "      <td>CHEMBL1000</td>\n",
       "      <td>Binding - unknown pharmacological activity</td>\n",
       "      <td>pK(i)</td>\n",
       "      <td>...</td>\n",
       "      <td>7.300</td>\n",
       "      <td>1.000</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[3H]-mepyramine (radioligand)</td>\n",
       "      <td>58.3</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>16408006</td>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>111</td>\n",
       "      <td>S</td>\n",
       "      <td>C</td>\n",
       "      <td>Cetirizine</td>\n",
       "      <td>ChEMBL Compound ID</td>\n",
       "      <td>CHEMBL1000</td>\n",
       "      <td>Binding - unknown pharmacological activity</td>\n",
       "      <td>pK(i)</td>\n",
       "      <td>...</td>\n",
       "      <td>7.500</td>\n",
       "      <td>1.585</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[3H]-mepyramine (radioligand)</td>\n",
       "      <td>55.0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>455</th>\n",
       "      <td>10.1124/mol.63.5.1021</td>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>274</td>\n",
       "      <td>N</td>\n",
       "      <td>A</td>\n",
       "      <td>VUF-5455</td>\n",
       "      <td>PubChem CID</td>\n",
       "      <td>10339071</td>\n",
       "      <td>Allosteric inverse agonist</td>\n",
       "      <td>k-1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.083</td>\n",
       "      <td>-3.322</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>I-AB-MECA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>=</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456</th>\n",
       "      <td>10.1124/mol.63.5.1021</td>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>274</td>\n",
       "      <td>N</td>\n",
       "      <td>A</td>\n",
       "      <td>I-AB-MECA</td>\n",
       "      <td>PubChem CID</td>\n",
       "      <td>44208896</td>\n",
       "      <td>Full agonist</td>\n",
       "      <td>k-1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.085</td>\n",
       "      <td>-1.328</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>=</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>457</th>\n",
       "      <td>10.1006/bbrc.2001.5027</td>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>282</td>\n",
       "      <td>Y</td>\n",
       "      <td>F</td>\n",
       "      <td>Cl-IB-MECA</td>\n",
       "      <td>PubChem CID</td>\n",
       "      <td>393593</td>\n",
       "      <td>Full agonist</td>\n",
       "      <td>EC50</td>\n",
       "      <td>...</td>\n",
       "      <td>27.600</td>\n",
       "      <td>-29.412</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>=</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458</th>\n",
       "      <td>10.1006/bbrc.2001.5027</td>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>282</td>\n",
       "      <td>Y</td>\n",
       "      <td>F</td>\n",
       "      <td>I-AB-MECA</td>\n",
       "      <td>PubChem CID</td>\n",
       "      <td>44208896</td>\n",
       "      <td>Full agonist</td>\n",
       "      <td>K(d)</td>\n",
       "      <td>...</td>\n",
       "      <td>9.300</td>\n",
       "      <td>-7.752</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>=</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>459</th>\n",
       "      <td>10.1006/bbrc.2001.5027</td>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>282</td>\n",
       "      <td>Y</td>\n",
       "      <td>F</td>\n",
       "      <td>Cl-IB-MECA</td>\n",
       "      <td>PubChem CID</td>\n",
       "      <td>393593</td>\n",
       "      <td>Full agonist</td>\n",
       "      <td>K(i)</td>\n",
       "      <td>...</td>\n",
       "      <td>33.600</td>\n",
       "      <td>-8.850</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>=</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11979 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  reference     protein  mutation_pos mutation_from  \\\n",
       "22                 15626750  hrh1_human            37             I   \n",
       "23                 15626750  hrh1_human            38             C   \n",
       "24                 15626750  hrh1_human            84             N   \n",
       "26                 16408006  hrh1_human           111             S   \n",
       "27                 16408006  hrh1_human           111             S   \n",
       "..                      ...         ...           ...           ...   \n",
       "455   10.1124/mol.63.5.1021  aa3r_human           274             N   \n",
       "456   10.1124/mol.63.5.1021  aa3r_human           274             N   \n",
       "457  10.1006/bbrc.2001.5027  aa3r_human           282             Y   \n",
       "458  10.1006/bbrc.2001.5027  aa3r_human           282             Y   \n",
       "459  10.1006/bbrc.2001.5027  aa3r_human           282             Y   \n",
       "\n",
       "    mutation_to ligand_name       ligand_idtype   ligand_id  \\\n",
       "22            V  Cetirizine  ChEMBL Compound ID  CHEMBL1000   \n",
       "23            S  Cetirizine  ChEMBL Compound ID  CHEMBL1000   \n",
       "24            S  Cetirizine  ChEMBL Compound ID  CHEMBL1000   \n",
       "26            A  Cetirizine  ChEMBL Compound ID  CHEMBL1000   \n",
       "27            C  Cetirizine  ChEMBL Compound ID  CHEMBL1000   \n",
       "..          ...         ...                 ...         ...   \n",
       "455           A    VUF-5455         PubChem CID    10339071   \n",
       "456           A   I-AB-MECA         PubChem CID    44208896   \n",
       "457           F  Cl-IB-MECA         PubChem CID      393593   \n",
       "458           F   I-AB-MECA         PubChem CID    44208896   \n",
       "459           F  Cl-IB-MECA         PubChem CID      393593   \n",
       "\n",
       "                                   ligand_class exp_type  ...  \\\n",
       "22   Binding - unknown pharmacological activity    pK(i)  ...   \n",
       "23   Binding - unknown pharmacological activity    pK(i)  ...   \n",
       "24   Binding - unknown pharmacological activity    pK(i)  ...   \n",
       "26   Binding - unknown pharmacological activity    pK(i)  ...   \n",
       "27   Binding - unknown pharmacological activity    pK(i)  ...   \n",
       "..                                          ...      ...  ...   \n",
       "455                  Allosteric inverse agonist      k-1  ...   \n",
       "456                                Full agonist      k-1  ...   \n",
       "457                                Full agonist     EC50  ...   \n",
       "458                                Full agonist     K(d)  ...   \n",
       "459                                Full agonist     K(i)  ...   \n",
       "\n",
       "    exp_mu_effect_value  exp_fold_change exp_mu_effect_qual  \\\n",
       "22                7.700           -1.585                      \n",
       "23                7.900            1.000                      \n",
       "24                7.500           -2.513                      \n",
       "26                7.300            1.000                      \n",
       "27                7.500            1.585                      \n",
       "..                  ...              ...                ...   \n",
       "455               0.083           -3.322                      \n",
       "456               0.085           -1.328                      \n",
       "457              27.600          -29.412                      \n",
       "458               9.300           -7.752                      \n",
       "459              33.600           -8.850                      \n",
       "\n",
       "    exp_mu_effect_ligand_prop              exp_mu_ligand_ref  \\\n",
       "22                             [3H]-mepyramine (radioligand)   \n",
       "23                             [3H]-mepyramine (radioligand)   \n",
       "24                             [3H]-mepyramine (radioligand)   \n",
       "26                             [3H]-mepyramine (radioligand)   \n",
       "27                             [3H]-mepyramine (radioligand)   \n",
       "..                        ...                            ...   \n",
       "455                                                I-AB-MECA   \n",
       "456                                                            \n",
       "457                                                            \n",
       "458                                                            \n",
       "459                                                            \n",
       "\n",
       "     opt_receptor_expression  opt_basal_activity opt_gain_of_activity  \\\n",
       "22                      94.2                   0                        \n",
       "23                     141.3                   0                        \n",
       "24                      53.7                   0                        \n",
       "26                      58.3                   0                        \n",
       "27                      55.0                   0                        \n",
       "..                       ...                 ...                  ...   \n",
       "455                        0                   0                    =   \n",
       "456                        0                   0                    =   \n",
       "457                        0                   0                    =   \n",
       "458                        0                   0                    =   \n",
       "459                        0                   0                    =   \n",
       "\n",
       "    opt_ligand_emax opt_agonist  \n",
       "22                0              \n",
       "23                0              \n",
       "24                0              \n",
       "26                0              \n",
       "27                0              \n",
       "..              ...         ...  \n",
       "455               0              \n",
       "456               0              \n",
       "457               0              \n",
       "458               0              \n",
       "459               0              \n",
       "\n",
       "[11979 rows x 25 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# we need to delete row with exp_fold_change==0 for both vep_input_all and gpcr_all_mutants\n",
    "# drop the lines with exp_fold_change==0 \n",
    "gpcr_filtered_mutants = gpcr_all_mutants[gpcr_all_mutants['exp_fold_change']!=0]\n",
    "gpcr_filtered_mutants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>reference</th>\n",
       "      <th>protein</th>\n",
       "      <th>mutation_pos</th>\n",
       "      <th>mutation_from</th>\n",
       "      <th>mutation_to</th>\n",
       "      <th>ligand_name</th>\n",
       "      <th>ligand_idtype</th>\n",
       "      <th>ligand_id</th>\n",
       "      <th>ligand_class</th>\n",
       "      <th>exp_type</th>\n",
       "      <th>...</th>\n",
       "      <th>exp_mu_effect_value</th>\n",
       "      <th>exp_fold_change</th>\n",
       "      <th>exp_mu_effect_qual</th>\n",
       "      <th>exp_mu_effect_ligand_prop</th>\n",
       "      <th>exp_mu_ligand_ref</th>\n",
       "      <th>opt_receptor_expression</th>\n",
       "      <th>opt_basal_activity</th>\n",
       "      <th>opt_gain_of_activity</th>\n",
       "      <th>opt_ligand_emax</th>\n",
       "      <th>opt_agonist</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>15626750</td>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>37</td>\n",
       "      <td>I</td>\n",
       "      <td>V</td>\n",
       "      <td>Cetirizine</td>\n",
       "      <td>ChEMBL Compound ID</td>\n",
       "      <td>CHEMBL1000</td>\n",
       "      <td>Binding - unknown pharmacological activity</td>\n",
       "      <td>pK(i)</td>\n",
       "      <td>...</td>\n",
       "      <td>7.70</td>\n",
       "      <td>-1.585</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[3H]-mepyramine (radioligand)</td>\n",
       "      <td>94.2</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>15626750</td>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>38</td>\n",
       "      <td>C</td>\n",
       "      <td>S</td>\n",
       "      <td>Cetirizine</td>\n",
       "      <td>ChEMBL Compound ID</td>\n",
       "      <td>CHEMBL1000</td>\n",
       "      <td>Binding - unknown pharmacological activity</td>\n",
       "      <td>pK(i)</td>\n",
       "      <td>...</td>\n",
       "      <td>7.90</td>\n",
       "      <td>1.000</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[3H]-mepyramine (radioligand)</td>\n",
       "      <td>141.3</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>15626750</td>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>84</td>\n",
       "      <td>N</td>\n",
       "      <td>S</td>\n",
       "      <td>Cetirizine</td>\n",
       "      <td>ChEMBL Compound ID</td>\n",
       "      <td>CHEMBL1000</td>\n",
       "      <td>Binding - unknown pharmacological activity</td>\n",
       "      <td>pK(i)</td>\n",
       "      <td>...</td>\n",
       "      <td>7.50</td>\n",
       "      <td>-2.513</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[3H]-mepyramine (radioligand)</td>\n",
       "      <td>53.7</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>16408006</td>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>111</td>\n",
       "      <td>S</td>\n",
       "      <td>A</td>\n",
       "      <td>Cetirizine</td>\n",
       "      <td>ChEMBL Compound ID</td>\n",
       "      <td>CHEMBL1000</td>\n",
       "      <td>Binding - unknown pharmacological activity</td>\n",
       "      <td>pK(i)</td>\n",
       "      <td>...</td>\n",
       "      <td>7.30</td>\n",
       "      <td>1.000</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[3H]-mepyramine (radioligand)</td>\n",
       "      <td>58.3</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>16408006</td>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>111</td>\n",
       "      <td>S</td>\n",
       "      <td>C</td>\n",
       "      <td>Cetirizine</td>\n",
       "      <td>ChEMBL Compound ID</td>\n",
       "      <td>CHEMBL1000</td>\n",
       "      <td>Binding - unknown pharmacological activity</td>\n",
       "      <td>pK(i)</td>\n",
       "      <td>...</td>\n",
       "      <td>7.50</td>\n",
       "      <td>1.585</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[3H]-mepyramine (radioligand)</td>\n",
       "      <td>55.0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346</th>\n",
       "      <td>10.1074/jbc.M110960200</td>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>247</td>\n",
       "      <td>S</td>\n",
       "      <td>A</td>\n",
       "      <td>I-AB-MECA</td>\n",
       "      <td>PubChem CID</td>\n",
       "      <td>44208896</td>\n",
       "      <td>Full agonist</td>\n",
       "      <td>K(d)</td>\n",
       "      <td>...</td>\n",
       "      <td>1.70</td>\n",
       "      <td>1.000</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>=</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>391</th>\n",
       "      <td>10.1074/jbc.M110960200</td>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>272</td>\n",
       "      <td>H</td>\n",
       "      <td>E</td>\n",
       "      <td>CADO</td>\n",
       "      <td>PubChem CID</td>\n",
       "      <td>8974</td>\n",
       "      <td>Full agonist</td>\n",
       "      <td>K(i)</td>\n",
       "      <td>...</td>\n",
       "      <td>11.60</td>\n",
       "      <td>-22.222</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>I-AB-MECA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>=</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433</th>\n",
       "      <td>10.1021/jm050968b</td>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>272</td>\n",
       "      <td>H</td>\n",
       "      <td>D</td>\n",
       "      <td>CID75022615</td>\n",
       "      <td>PubChem CID</td>\n",
       "      <td>75022615</td>\n",
       "      <td>Full agonist</td>\n",
       "      <td>K(i)</td>\n",
       "      <td>...</td>\n",
       "      <td>0.12</td>\n",
       "      <td>1.167</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>I-AB-MECA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>=</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>446</th>\n",
       "      <td>10.1124/mol.63.5.1021</td>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>274</td>\n",
       "      <td>N</td>\n",
       "      <td>A</td>\n",
       "      <td>I-AB-MECA</td>\n",
       "      <td>PubChem CID</td>\n",
       "      <td>44208896</td>\n",
       "      <td>Full agonist</td>\n",
       "      <td>K(d)</td>\n",
       "      <td>...</td>\n",
       "      <td>4.60</td>\n",
       "      <td>-3.831</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>=</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>457</th>\n",
       "      <td>10.1006/bbrc.2001.5027</td>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>282</td>\n",
       "      <td>Y</td>\n",
       "      <td>F</td>\n",
       "      <td>Cl-IB-MECA</td>\n",
       "      <td>PubChem CID</td>\n",
       "      <td>393593</td>\n",
       "      <td>Full agonist</td>\n",
       "      <td>EC50</td>\n",
       "      <td>...</td>\n",
       "      <td>27.60</td>\n",
       "      <td>-29.412</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>=</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3071 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  reference     protein  mutation_pos mutation_from  \\\n",
       "22                 15626750  hrh1_human            37             I   \n",
       "23                 15626750  hrh1_human            38             C   \n",
       "24                 15626750  hrh1_human            84             N   \n",
       "26                 16408006  hrh1_human           111             S   \n",
       "27                 16408006  hrh1_human           111             S   \n",
       "..                      ...         ...           ...           ...   \n",
       "346  10.1074/jbc.M110960200  aa3r_human           247             S   \n",
       "391  10.1074/jbc.M110960200  aa3r_human           272             H   \n",
       "433       10.1021/jm050968b  aa3r_human           272             H   \n",
       "446   10.1124/mol.63.5.1021  aa3r_human           274             N   \n",
       "457  10.1006/bbrc.2001.5027  aa3r_human           282             Y   \n",
       "\n",
       "    mutation_to  ligand_name       ligand_idtype   ligand_id  \\\n",
       "22            V   Cetirizine  ChEMBL Compound ID  CHEMBL1000   \n",
       "23            S   Cetirizine  ChEMBL Compound ID  CHEMBL1000   \n",
       "24            S   Cetirizine  ChEMBL Compound ID  CHEMBL1000   \n",
       "26            A   Cetirizine  ChEMBL Compound ID  CHEMBL1000   \n",
       "27            C   Cetirizine  ChEMBL Compound ID  CHEMBL1000   \n",
       "..          ...          ...                 ...         ...   \n",
       "346           A    I-AB-MECA         PubChem CID    44208896   \n",
       "391           E         CADO         PubChem CID        8974   \n",
       "433           D  CID75022615         PubChem CID    75022615   \n",
       "446           A    I-AB-MECA         PubChem CID    44208896   \n",
       "457           F   Cl-IB-MECA         PubChem CID      393593   \n",
       "\n",
       "                                   ligand_class exp_type  ...  \\\n",
       "22   Binding - unknown pharmacological activity    pK(i)  ...   \n",
       "23   Binding - unknown pharmacological activity    pK(i)  ...   \n",
       "24   Binding - unknown pharmacological activity    pK(i)  ...   \n",
       "26   Binding - unknown pharmacological activity    pK(i)  ...   \n",
       "27   Binding - unknown pharmacological activity    pK(i)  ...   \n",
       "..                                          ...      ...  ...   \n",
       "346                                Full agonist     K(d)  ...   \n",
       "391                                Full agonist     K(i)  ...   \n",
       "433                                Full agonist     K(i)  ...   \n",
       "446                                Full agonist     K(d)  ...   \n",
       "457                                Full agonist     EC50  ...   \n",
       "\n",
       "    exp_mu_effect_value  exp_fold_change exp_mu_effect_qual  \\\n",
       "22                 7.70           -1.585                      \n",
       "23                 7.90            1.000                      \n",
       "24                 7.50           -2.513                      \n",
       "26                 7.30            1.000                      \n",
       "27                 7.50            1.585                      \n",
       "..                  ...              ...                ...   \n",
       "346                1.70            1.000                      \n",
       "391               11.60          -22.222                      \n",
       "433                0.12            1.167                      \n",
       "446                4.60           -3.831                      \n",
       "457               27.60          -29.412                      \n",
       "\n",
       "    exp_mu_effect_ligand_prop              exp_mu_ligand_ref  \\\n",
       "22                             [3H]-mepyramine (radioligand)   \n",
       "23                             [3H]-mepyramine (radioligand)   \n",
       "24                             [3H]-mepyramine (radioligand)   \n",
       "26                             [3H]-mepyramine (radioligand)   \n",
       "27                             [3H]-mepyramine (radioligand)   \n",
       "..                        ...                            ...   \n",
       "346                                                            \n",
       "391                                                I-AB-MECA   \n",
       "433                                                I-AB-MECA   \n",
       "446                                                            \n",
       "457                                                            \n",
       "\n",
       "     opt_receptor_expression  opt_basal_activity opt_gain_of_activity  \\\n",
       "22                      94.2                   0                        \n",
       "23                     141.3                   0                        \n",
       "24                      53.7                   0                        \n",
       "26                      58.3                   0                        \n",
       "27                      55.0                   0                        \n",
       "..                       ...                 ...                  ...   \n",
       "346                        0                   0                    =   \n",
       "391                        0                   0                    =   \n",
       "433                        0                   0                    =   \n",
       "446                        0                   0                    =   \n",
       "457                        0                   0                    =   \n",
       "\n",
       "    opt_ligand_emax opt_agonist  \n",
       "22                0              \n",
       "23                0              \n",
       "24                0              \n",
       "26                0              \n",
       "27                0              \n",
       "..              ...         ...  \n",
       "346               0              \n",
       "391               0              \n",
       "433               0              \n",
       "446               0              \n",
       "457               0              \n",
       "\n",
       "[3071 rows x 25 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unq_gpcr = gpcr_filtered_mutants.drop_duplicates(subset=['protein','mutation_pos','mutation_from','mutation_to'], keep='first', inplace=False)\n",
    "unq_gpcr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein</th>\n",
       "      <th>ENST</th>\n",
       "      <th>mutation_pos</th>\n",
       "      <th>exp_fold_change</th>\n",
       "      <th>mutation_from</th>\n",
       "      <th>mutation_to</th>\n",
       "      <th>vep_input</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>ENST00000438284</td>\n",
       "      <td>37</td>\n",
       "      <td>-1.585</td>\n",
       "      <td>Ile</td>\n",
       "      <td>Val</td>\n",
       "      <td>ENST00000438284:p.Ile37Val</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>ENST00000438284</td>\n",
       "      <td>38</td>\n",
       "      <td>1.000</td>\n",
       "      <td>Cys</td>\n",
       "      <td>Ser</td>\n",
       "      <td>ENST00000438284:p.Cys38Ser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>ENST00000438284</td>\n",
       "      <td>84</td>\n",
       "      <td>-2.513</td>\n",
       "      <td>Asn</td>\n",
       "      <td>Ser</td>\n",
       "      <td>ENST00000438284:p.Asn84Ser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>ENST00000438284</td>\n",
       "      <td>111</td>\n",
       "      <td>1.000</td>\n",
       "      <td>Ser</td>\n",
       "      <td>Ala</td>\n",
       "      <td>ENST00000438284:p.Ser111Ala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>hrh1_human</td>\n",
       "      <td>ENST00000438284</td>\n",
       "      <td>111</td>\n",
       "      <td>1.585</td>\n",
       "      <td>Ser</td>\n",
       "      <td>Cys</td>\n",
       "      <td>ENST00000438284:p.Ser111Cys</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>455</th>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>ENST00000241356</td>\n",
       "      <td>274</td>\n",
       "      <td>-3.322</td>\n",
       "      <td>Asn</td>\n",
       "      <td>Ala</td>\n",
       "      <td>ENST00000241356:p.Asn274Ala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456</th>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>ENST00000241356</td>\n",
       "      <td>274</td>\n",
       "      <td>-1.328</td>\n",
       "      <td>Asn</td>\n",
       "      <td>Ala</td>\n",
       "      <td>ENST00000241356:p.Asn274Ala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>457</th>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>ENST00000241356</td>\n",
       "      <td>282</td>\n",
       "      <td>-29.412</td>\n",
       "      <td>Tyr</td>\n",
       "      <td>Phe</td>\n",
       "      <td>ENST00000241356:p.Tyr282Phe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458</th>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>ENST00000241356</td>\n",
       "      <td>282</td>\n",
       "      <td>-7.752</td>\n",
       "      <td>Tyr</td>\n",
       "      <td>Phe</td>\n",
       "      <td>ENST00000241356:p.Tyr282Phe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>459</th>\n",
       "      <td>aa3r_human</td>\n",
       "      <td>ENST00000241356</td>\n",
       "      <td>282</td>\n",
       "      <td>-8.850</td>\n",
       "      <td>Tyr</td>\n",
       "      <td>Phe</td>\n",
       "      <td>ENST00000241356:p.Tyr282Phe</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11979 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        protein             ENST  mutation_pos  exp_fold_change mutation_from  \\\n",
       "22   hrh1_human  ENST00000438284            37           -1.585           Ile   \n",
       "23   hrh1_human  ENST00000438284            38            1.000           Cys   \n",
       "24   hrh1_human  ENST00000438284            84           -2.513           Asn   \n",
       "26   hrh1_human  ENST00000438284           111            1.000           Ser   \n",
       "27   hrh1_human  ENST00000438284           111            1.585           Ser   \n",
       "..          ...              ...           ...              ...           ...   \n",
       "455  aa3r_human  ENST00000241356           274           -3.322           Asn   \n",
       "456  aa3r_human  ENST00000241356           274           -1.328           Asn   \n",
       "457  aa3r_human  ENST00000241356           282          -29.412           Tyr   \n",
       "458  aa3r_human  ENST00000241356           282           -7.752           Tyr   \n",
       "459  aa3r_human  ENST00000241356           282           -8.850           Tyr   \n",
       "\n",
       "    mutation_to                    vep_input  \n",
       "22          Val   ENST00000438284:p.Ile37Val  \n",
       "23          Ser   ENST00000438284:p.Cys38Ser  \n",
       "24          Ser   ENST00000438284:p.Asn84Ser  \n",
       "26          Ala  ENST00000438284:p.Ser111Ala  \n",
       "27          Cys  ENST00000438284:p.Ser111Cys  \n",
       "..          ...                          ...  \n",
       "455         Ala  ENST00000241356:p.Asn274Ala  \n",
       "456         Ala  ENST00000241356:p.Asn274Ala  \n",
       "457         Phe  ENST00000241356:p.Tyr282Phe  \n",
       "458         Phe  ENST00000241356:p.Tyr282Phe  \n",
       "459         Phe  ENST00000241356:p.Tyr282Phe  \n",
       "\n",
       "[11979 rows x 7 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vep_filtered_all = vep_input_all[vep_input_all['exp_fold_change']!=0]\n",
    "vep_filtered_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000019103:p.Ala210Cys</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000019103:p.Ala213Cys</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000019103:p.Ala290Cys</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000019103:p.Ala292Cys</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000019103:p.Ala359Cys</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3066</th>\n",
       "      <td>ENST00000646607:p.Lys273Ala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3067</th>\n",
       "      <td>ENST00000646607:p.Trp336Ala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3068</th>\n",
       "      <td>ENST00000646607:p.Tyr247Ala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3069</th>\n",
       "      <td>ENST00000646607:p.Tyr350Ala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3070</th>\n",
       "      <td>ENST00000646607:p.Tyr369Ala</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3071 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                0\n",
       "0     ENST00000019103:p.Ala210Cys\n",
       "1     ENST00000019103:p.Ala213Cys\n",
       "2     ENST00000019103:p.Ala290Cys\n",
       "3     ENST00000019103:p.Ala292Cys\n",
       "4     ENST00000019103:p.Ala359Cys\n",
       "...                           ...\n",
       "3066  ENST00000646607:p.Lys273Ala\n",
       "3067  ENST00000646607:p.Trp336Ala\n",
       "3068  ENST00000646607:p.Tyr247Ala\n",
       "3069  ENST00000646607:p.Tyr350Ala\n",
       "3070  ENST00000646607:p.Tyr369Ala\n",
       "\n",
       "[3071 rows x 1 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ll_vep = pd.DataFrame(np.unique(vep_filtered_all['vep_input'].tolist()))\n",
    "#drop 'nan\n",
    "# ll_vep = ll_vep[ll_vep[0]!='nan']\n",
    "ll_vep"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# output data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "vep_filtered_all.to_csv('vep_filtered_all.csv',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save the filtered vep inputs\n",
    "ll_vep.to_csv('all_vep_inputs.csv',index=False, header=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save to csv\n",
    "gpcr_filtered_mutants.to_csv(\"gpcr_filtered_mutants_0217.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "773d181160c386017ad51b585d282f9f41a012296bacf0d61cb3b8d5443ae20a"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
